---
title: "Baseline Kinome Analysis"
output: html_document
date: "`r Sys.Date()`"
---
```{r load libraries, echo = FALSE, include=FALSE}
if (!require("pacman")) install.packages("pacman")
pacman::p_load(org.Hs.eg.db, dplyr, tidyverse, fuzzyjoin, stringr, openxlsx, writexl,limma, 
               ggpubr, corrplot, PCAtools, ComplexHeatmap, EnhancedVolcano, ConsensusClusterPlus, circlize,
               rstatix, mclust, clusterProfiler, conflicted, here)

conflict_prefer("filter","dplyr")
conflict_prefer("select", "dplyr")
conflict_prefer("biplot", "PCAtools")
conflict_prefer("slice", "dplyr")

options(dplyr.summarise.inform = FALSE)
```

```{r set directory, include=FALSE}
path1 = "/Volumes/labs/Home/Jen Jen Yeh Lab/Yi/Projects/MIB/Analysis2/baseline_analysis"
path2 = "P:/Jen Jen Yeh Lab/Yi/Projects/MIB/Analysis2/baseline_analysis"
path3 = "Z:/Yi/Projects/MIB/Analysis2/baseline_analysis"

if (dir.exists(path1)) {
  knitr::opts_knit$set(root.dir = path1)
} else if (dir.exists(path2)) {
  knitr::opts_knit$set(root.dir = path2)
} else if (dir.exists(path3)) {
  knitr::opts_knit$set(root.dir = path3)
} else {
  stop("None of the specified paths exist. Please check the paths.")
}

knitr::opts_chunk$set(echo = TRUE)
knitr::opts_chunk$set(warning = FALSE)

#create directories
folders <- c("plots","tables")

# Loop through each folder name
for (folder in folders) {
  # Check if the folder exists, and create it if it does not
  if (!dir.exists(folder)) {
    dir.create(folder)
  }
}
```


```{r load Rds and functions, echo = FALSE, include=FALSE}
kinome_rds <- readRDS("kinase.rds")
blca_rds <- readRDS("blca.rds")
brca_rds <- readRDS("brca.rds")
pdac_rds <- readRDS("pdac.rds")

YehKinome <- c(kinome_rds, blca_rds, brca_rds, pdac_rds)

source("functions.R")

```

```{r color keys, echo = FALSE, include=FALSE}
col_SSC = c("classical" = "#0331fc", "basal" = "#fc8803")


col_kinotypes = c("k1" = "#B15928", "k2" = "#6A3D9A", "k3" ="#f4cc03")
col_line = c(         "P319T1" = "#1F77B4",
                      "P710T1" = "#FF7F0E",
                      "P203T1" = "#2CA02C",
                      "P616T1" = "#D62728",
                      "P109T1" = "#9467BD",
                      "P411T1" = "#8C564B",
                      "P125T2" = "#E377C2",
                      "P119T1" = "#7F7F7F",
                      "P225T1" = "#BCBD22",
                      "PancT6" = "#17BECF",
                      "P508T1" = "#AEC7E8",
                      "P902T1B" = "#FFBB78",
                      "P806T1" = "#98DF8A",
                      "P713T1" = "#FF9896"
             )

col_run = c("A" = "#1F77B4", 
            "B" = "#FF7F0E", 
            "C" = "#2CA02C", 
            "D" = "#D62728",
            "E" = "#9467BD", 
            "G" = "#8C564B", 
            "H" = "#E377C2", 
            "I" = "#7F7F7F", 
            "J" = "#BCBD22", 
            "K" = "#17BECF", 
            "L" = "#AEC7E8")

col_k10 = c(          "1" = "#393B79FF",
                      "2" = "#637939FF",
                      "3" = "#8C6D31FF",
                      "4" = "#843C39FF",
                      "5" = "#7B4173FF",
                      "6" = "#BCBDDCFF",
                      "7" = "#5254A3FF",
                      "8" = "#8CA252FF",
                      "9" = "#BD9E39FF",
                      "10" = "#AD494AFF",
                      "11" = "#A55194FF",
                      "12" = "#6B6ECFFF",
                      "13" = "#B5CF6BFF",
                      "14" = "#E7BA52FF"
             )

col_k14 = c(          "1" = "#393B79FF",
                      "2" = "#637939FF",
                      "3" = "#8C6D31FF",
                      "4" = "#843C39FF",
                      "5" = "#7B4173FF",
                      "6" = "#BCBDDCFF",
                      "7" = "#5254A3FF",
                      "8" = "#8CA252FF",
                      "9" = "#BD9E39FF",
                      "10" = "#AD494AFF",
                      "11" = "#A55194FF",
                      "12" = "#6B6ECFFF",
                      "13" = "#B5CF6BFF",
                      "14" = "#E7BA52FF"
             )

col_brca <- c("LumA" = "royalblue", 
                  "LumB" = "navy", 
                  "Her2" = "purple", 
                  "Basal" = "#fc8803", 
                  "Normal-like" = "black",
             "Luminal" = "blue")

col_fun = colorRamp2(c(0, 1), c("white", "blue"))

col_fun_htmp = colorRamp2(c(-2, 0, 2), c("blue", "white", "brown"))

```

```{r impute missing values, echo = FALSE, include=FALSE}
#load data and make rownnames
LFQ <- YehKinome$df_LFQ
LFQ <- data.frame(LFQ [,-1], row.names = LFQ [,1])

#identify line names, subset each line's replicates into list of dataframes
ids <- unique(sub("_.*", "", colnames(LFQ[,1:101])))

LFQ_list <- sapply(ids,
        function(x) LFQ[startsWith(names(LFQ),x)],
       simplify = FALSE)

LFQ_3reps <- list(P508T1 = LFQ_list$P508T1, P902T1B = LFQ_list$P902T1B
                  , P806T1 = LFQ_list$P806T1)
LFQ_4reps <- list(P713T1 = LFQ_list$P713T1)
LFQ_5reps <- list(P225T1 = LFQ_list$P225T1)
LFQ_6reps <- list(P203T1 = LFQ_list$P203T1, P109T1 = LFQ_list$P109T1,
                  P125T2 = LFQ_list$P125T2)
LFQ_9reps <- list(P319T1 = LFQ_list$P319T1, P710T1 = LFQ_list$P710T1, P616T1 = LFQ_list$P616T1)
LFQ_12reps <- list(PancT6 = LFQ_list$PancT6, P119T1 = LFQ_list$P119T1)
LFQ_14reps <- list(P411T1 = LFQ_list$P411T1)


res3 <- lapply(LFQ_3reps, impute3)
res4 <- lapply(LFQ_4reps, impute4)
res5 <- lapply(LFQ_5reps, impute5)
res6 <- lapply(LFQ_6reps, impute6)
res9 <- lapply(LFQ_9reps, impute9)
res12 <- lapply(LFQ_12reps, impute12)
res14 <- lapply(LFQ_14reps, impute14)

res <- c(res3, res4, res5, res6, res9, res12, res14)
res <- res[ids]
res <- bind_cols(res)
res <- cbind(res, LFQ[,102:111])

col.order <- colnames(LFQ)
df_impute_line <- res[,col.order]

```

```{r define data and metadata, echo = FALSE, include=FALSE}
##metadata
#all metadata
m <- YehKinome$m

#metadata without pool
m2 <- YehKinome$m2

#collapsed metadata by line
m_line <- m
m_line$ID <- sub("_[^_]+$", "", m$ID) #remove PRE1, PRE2 etc
m_line$ID <- gsub("(Pool).*", "\\1", m_line$ID) #remove pool ID
ID_order <- unique(m_line$ID) #define order of samples
concat_unique <- function(x){paste(unique(x),  collapse=',')} #fn for collapsing
m_line <- m_line %>% 
  group_by(ID) %>% 
  summarise_all(concat_unique)
m_line <- m_line[match(ID_order, m_line$ID),]

#collapsed metadata no pool
m_line2 <- m_line[1:14,]

#collapsed metadata by subtype
m_line2$basal <- as.numeric(m_line2$basal)
m_line2$classical <- as.numeric(m_line2$classical)

#RNA metadata
m_rna <- YehKinome$m_rna

##dataframes
#proteinGroup by species
all <- YehKinome$df_raw

#LFQ
LFQ <- YehKinome$df_LFQ
LFQ <- data.frame(LFQ [,-1], row.names = LFQ [,1])

#LFQ with original colnames
LFQ_raw <- YehKinome$df_LFQ_raw
LFQ_raw <- data.frame(LFQ_raw [,-1], row.names = LFQ_raw [,1])

#imputed LFQ by mean of each line
LFQ_imp <- df_impute_line

#collapsed LFQ by mean of each line
LFQ_c <- data.frame(t(LFQ_raw))
LFQ_c$line <- m$line

LFQ_c <- LFQ_c %>% relocate(line, .before = PDPK1) %>% 
  group_by(line) %>% 
  summarise(across(where(is.numeric), mean))

LFQ_c <- data.frame((LFQ_c))
LFQ_c <- LFQ_c[match(m_line$line, LFQ_c$line),]
rownames(LFQ_c) <- LFQ_c$line
LFQ_c$line <- NULL
LFQ_c <- data.frame(t(LFQ_c))
LFQ_c[LFQ_c == 0] <- NA

#RNAseq
TPM <- YehKinome$df_pdxrna$ex

TPM_ENSG <- TPM
rownames(TPM_ENSG) <- YehKinome$df_pdxrna$featInfo$ENSG
rownames(TPM_ENSG) <- gsub("\\..*","",rownames(TPM_ENSG))

##genesets
TSP_geneset <- YehKinome$TSP_geneset
collisson_geneset <- YehKinome$collisson_geneset
bailey_geneset <- YehKinome$bailey_geneset
moffitt_geneset <- YehKinome$moffitt_geneset

```


```{r normalize data, echo = FALSE, include=FALSE}
#LFQ imputed
d <- LFQ_imp[,1:101]
d$PoolAvg <- rowMeans(subset(LFQ_imp, select = c(PoolA, PoolB, PoolC, PoolD, PoolE, PoolG, PoolH, PoolI, PoolJ, PoolL)), na.rm = F)
summary(d)
d[d == 0] <- NA
d.log2 <- log2(d)
d.log2.rb <- d.log2[,1:101] - d.log2[,102]
cmeds_d <- apply(d.log2.rb, 2, medianWithoutNA)
d.log2.rb.MC <- sweep(d.log2.rb, 2, cmeds_d, "-")

#LFQ collapsed
dc.log2 <- log2(LFQ_c)
dc.log2.rb <- dc.log2[1:14] - dc.log2[,15]
cmeds_dc <- apply(dc.log2.rb, 2, medianWithoutNA)
dc.log2.rb.MC <- sweep(dc.log2.rb, 2, cmeds_dc, "-")

#RNAseq
RNA_ex <- log2(TPM+1)
RNA_ex_ENSG <- log2(TPM_ENSG+1)

```

```{r kinase stats data, echo = FALSE, include=FALSE}
all_kinases <- all$Kinases
human_kinases <- all$Human
mouse_kinases <- all$Mouse
amb_kinases <- all$Ambiguous

all_kinases_LFQ <- all_kinases[ , grepl( "LFQ" , names( all_kinases ) ) ]
human_kinases_LFQ <- human_kinases[ , grepl( "LFQ" , names( human_kinases ) ) ]
mouse_kinases_LFQ <- mouse_kinases[ , grepl( "LFQ" , names( mouse_kinases ) ) ]
amb_kinases_LFQ <- amb_kinases[ , grepl( "LFQ" , names( amb_kinases ) ) ]

all_kinases_LFQ <- all_kinases_LFQ[,-55]
human_kinases_LFQ <- human_kinases_LFQ[,-55]
mouse_kinases_LFQ <- mouse_kinases_LFQ[,-55]
amb_kinases_LFQ <- amb_kinases_LFQ[,-55]

LFQ_samp_key <- data.frame(LFQ = gsub(".", " ", colnames(LFQ_raw), fixed = TRUE), name = colnames(LFQ))
LFQ_samp_key[99,1] <- "LFQ intensity SB19-2"
LFQ_samp_key[100,1] <- "LFQ intensity SB19-3"
LFQ_samp_key[101,1] <- "LFQ intensity SB19-4"

colnames(all_kinases_LFQ) <- dplyr::recode(
  colnames(all_kinases_LFQ), 
  !!!setNames(as.character(LFQ_samp_key$name), LFQ_samp_key$LFQ)
)

colnames(human_kinases_LFQ) <- dplyr::recode(
  colnames(human_kinases_LFQ), 
  !!!setNames(as.character(LFQ_samp_key$name), LFQ_samp_key$LFQ)
)

colnames(mouse_kinases_LFQ) <- dplyr::recode(
  colnames(mouse_kinases_LFQ), 
  !!!setNames(as.character(LFQ_samp_key$name), LFQ_samp_key$LFQ)
)

colnames(amb_kinases_LFQ) <- dplyr::recode(
  colnames(amb_kinases_LFQ), 
  !!!setNames(as.character(LFQ_samp_key$name), LFQ_samp_key$LFQ)
)

all_kinases_LFQ[all_kinases_LFQ == 0] <- NA
human_kinases_LFQ[human_kinases_LFQ == 0] <- NA
mouse_kinases_LFQ[mouse_kinases_LFQ == 0] <- NA
amb_kinases_LFQ[amb_kinases_LFQ == 0] <- NA

all_kinase_cnts <- data.frame(colSums(!is.na(all_kinases_LFQ)))
colnames(all_kinase_cnts) <- "counts"

m_kinase_cnts <- data.frame(colSums(!is.na(mouse_kinases_LFQ)))
colnames(m_kinase_cnts) <- "counts"

a_kinase_cnts <- data.frame(colSums(!is.na(amb_kinases_LFQ)))
colnames(a_kinase_cnts) <- "counts"

h_kinase_cnts <- data.frame(colSums(!is.na(human_kinases_LFQ)))
colnames(h_kinase_cnts) <- "counts"

m_kinase_cnts <- data.frame(colSums(!is.na(mouse_kinases_LFQ)))
colnames(m_kinase_cnts) <- "counts"

a_kinase_cnts <- data.frame(colSums(!is.na(amb_kinases_LFQ)))
colnames(a_kinase_cnts) <- "counts"

#get line names
all_kinase_cnts$line <- gsub("_.*", "", gsub("\\l[^.]*$","l", rownames(all_kinase_cnts)))
h_kinase_cnts$line <- gsub("_.*", "", gsub("\\l[^.]*$","l", rownames(h_kinase_cnts)))
m_kinase_cnts$line <- gsub("_.*", "", gsub("\\l[^.]*$","l", rownames(m_kinase_cnts)))
a_kinase_cnts$line <- gsub("_.*", "", gsub("\\l[^.]*$","l", rownames(a_kinase_cnts)))

#get basal/classical
all_kinase_cnts$species <- "all"
h_kinase_cnts$species <- "human"
m_kinase_cnts$species <- "mouse"
a_kinase_cnts$species <- "ambiguous"

all_kinase_cnts$subtype <- m$subtype[match(all_kinase_cnts$line, m2$line)]
h_kinase_cnts$subtype <- m$subtype[match(h_kinase_cnts$line, m2$line)]
m_kinase_cnts$subtype <- m$subtype[match(m_kinase_cnts$line, m2$line)]
a_kinase_cnts$subtype <- m$subtype[match(a_kinase_cnts$line, m2$line)]

#plotting out cnts before averaging out, allows for error bars to be plotted
complete_kinase_cnts <- bind_rows(h_kinase_cnts, m_kinase_cnts, a_kinase_cnts)

levels(as.factor(complete_kinase_cnts$species))
complete_kinase_cnts$species <- factor(complete_kinase_cnts$species, levels = c("human", "mouse", "ambiguous")) 

levels(as.factor(complete_kinase_cnts$subtype))
complete_kinase_cnts$subtype <- factor(complete_kinase_cnts$subtype, levels = c("basal", "classical")) 

#average by line
h_kinase_cnts_avg <- h_kinase_cnts %>%
  group_by(line, species) %>%
  summarise(kinases=(mean(counts)))

m_kinase_cnts_avg <- m_kinase_cnts %>%
  group_by(line, species) %>%
  summarise(kinases=(mean(counts)))

a_kinase_cnts_avg <- a_kinase_cnts %>%
  group_by(line, species) %>%
  summarise(kinases=(mean(counts)))

#frequency per species
all_kinases_LFQ2 <- all_kinases_LFQ
human_kinases_LFQ2 <- human_kinases_LFQ
mouse_kinases_LFQ2 <- mouse_kinases_LFQ
amb_kinases_LFQ2 <- amb_kinases_LFQ

all_kinases_LFQ2$countNA <- rowSums(is.na(all_kinases_LFQ2[1:111]))
human_kinases_LFQ2$countNA <- rowSums(is.na(human_kinases_LFQ2[1:111]))
mouse_kinases_LFQ2$countNA <- rowSums(is.na(mouse_kinases_LFQ[1:111]))
amb_kinases_LFQ2$countNA <- rowSums(is.na(amb_kinases_LFQ2[1:111]))

freq75 <- all_kinases_LFQ2
freq75[with(freq75, freq75$countNA >28),] <- NA
k75 <- which(is.na(freq75[1:111]), arr.ind=TRUE)
freq75[k75] <- rowMeans(freq75[1:111], na.rm=TRUE)[k75[,1]]
freq75 <- freq75[,1:111]
freq75 <- na.omit(freq75)

freq50 <- all_kinases_LFQ2
freq50[with(freq50, freq50$countNA >56),] <- NA
k50 <- which(is.na(freq50[1:111]), arr.ind=TRUE)
freq50[k50] <- rowMeans(freq50[1:111], na.rm=TRUE)[k50[,1]]
freq50 <- freq50[,1:111]
freq50 <- na.omit(freq50)

freq75_h <- human_kinases_LFQ2
freq75_h[with(freq75_h, freq75_h$countNA >28),] <- NA
k75_h <- which(is.na(freq75_h[1:111]), arr.ind=TRUE)
freq75_h[k75_h] <- rowMeans(freq75_h[1:111], na.rm=TRUE)[k75_h[,1]]
freq75_h <- freq75_h[,1:111]
freq75_h <- na.omit(freq75_h)

freq50_h <- human_kinases_LFQ2
freq50_h[with(freq50_h, freq50_h$countNA >56),] <- NA
k50_h <- which(is.na(freq50_h[1:111]), arr.ind=TRUE)
freq50_h[k50_h] <- rowMeans(freq50_h[1:111], na.rm=TRUE)[k50_h[,1]]
freq50_h <- freq50_h[,1:111]
freq50_h <- na.omit(freq50_h)

freq75_m <- mouse_kinases_LFQ2
freq75_m[with(freq75_m, freq75_m$countNA >28),] <- NA
k75_m <- which(is.na(freq75_m[1:111]), arr.ind=TRUE)
freq75_m[k75_m] <- rowMeans(freq75_m[1:111], na.rm=TRUE)[k75_m[,1]]
freq75_m <- freq75_m[,1:111]
freq75_m <- na.omit(freq75_m)

freq50_m <- mouse_kinases_LFQ2
freq50_m[with(freq50_m, freq50_m$countNA >56),] <- NA
k50_m <- which(is.na(freq50_m[1:69]), arr.ind=TRUE)
freq50_m[k50_m] <- rowMeans(freq50_m[1:69], na.rm=TRUE)[k50_m[,1]]
freq50_m <- freq50_m[,1:69]
freq50_m <- na.omit(freq50_m)

freq75_a <- amb_kinases_LFQ2
freq75_a[with(freq75_a, freq75_a$countNA >28),] <- NA
k75_a <- which(is.na(freq75_a[1:111]), arr.ind=TRUE)
freq75_a[k75_a] <- rowMeans(freq75_a[1:111], na.rm=TRUE)[k75_a[,1]]
freq75_a <- freq75_a[,1:111]
freq75_a <- na.omit(freq75_a)

freq50_a <- amb_kinases_LFQ2
freq50_a[with(freq50_a, freq50_a$countNA >56),] <- NA
k50_a <- which(is.na(freq50_a[1:111]), arr.ind=TRUE)
freq50_a[k50_a] <- rowMeans(freq50_a[1:111], na.rm=TRUE)[k50_a[,1]]
freq50_a <- freq50_a[,1:111]
freq50_a <- na.omit(freq50_a)

kinase_total <- nrow(all_kinases_LFQ)
kinase_freq75 <- nrow(freq75)
kinase_freq50 <- nrow(freq50)
kinase_h <- nrow(human_kinases_LFQ)
kinase_h_freq75 <- nrow(freq75_h)
kinase_h_freq50 <- nrow(freq50_h)
kinase_m <- nrow(mouse_kinases_LFQ)
kinase_m_freq75 <- nrow(freq75_m)
kinase_m_freq50 <- nrow(freq50_m)
kinase_a <- nrow(amb_kinases_LFQ2)
kinase_a_freq75 <- nrow(freq75_a)
kinase_a_freq50 <- nrow(freq50_a)

group <- c("all", "all", "all", "human","human","human", "mouse","mouse","mouse","ambiguous","ambiguous","ambiguous")
frequency <- c("total", ">75%", ">50%","total", ">75%", ">50%","total", ">75%", ">50%","total", ">75%", ">50%")
count <- c(kinase_total,kinase_freq75,kinase_freq50,kinase_h,kinase_h_freq75,kinase_h_freq50,kinase_m
           ,kinase_m_freq75,kinase_m_freq50,kinase_a,kinase_a_freq75,kinase_a_freq50)

frequency_df <- data.frame(group, frequency, count)
frequency_df$frequency <- factor(frequency_df$frequency, levels = c("total", ">50%", ">75%"))

```

```{r kinase stats plots, fig.height=6, fig.width=9, echo = FALSE, message = FALSE, error=TRUE}
fig1b <- ggbarplot(complete_kinase_cnts, "line", "counts",
          fill = "species", palette = c("#2171b5","#6baed6","#bdd7e7"), width = 0.85,
          add = "mean_sd",
          label = T, lab.col = "black", lab.size = 5.5,
          order = rev(a_kinase_cnts_avg$line),
          ylab = "kinases",
          lab.nb.digits = 0,
          lab.pos = "in") + theme(axis.text.x = element_text(size = 18 ,angle = 45, vjust = 0.8, hjust=0.8)) +  font("xlab", size = 18, face = "bold") + font("ylab", size = 18, face = "bold") + font("legend.title", color = "black", face = "bold", size = 18)+ 
  font("legend.text", color = "black", face = "bold", size = 20) + font("y.text", color = "black", size = 18)

fig1c <- ggbarplot(frequency_df, x = "group", y = "count",
          fill = "frequency", width = 0.8,
          palette = c("#2171b5","#6baed6","#bdd7e7"),
          label = TRUE, lab.col = "black", lab.size = 6,
          position = position_dodge(0.9)) + theme(axis.text.x = element_text(size = 20 , vjust = 0.8)) +  font("xlab", size = 20, face = "bold") + font("ylab", size = 20, face = "bold") + font("legend.title", color = "black", face = "bold", size = 20)+ 
  font("legend.text", color = "black", face = "bold", size = 20) + font("y.text", color = "black", size = 20)+ylim(0,500)

png("plots/fig1b.png",width=9,height=6,units="in",res=500)
fig1b
invisible(dev.off())


png("plots/fig1c.png",width=9,height=6.2,units="in",res=500)
fig1c
invisible(dev.off())

fig1b

fig1c

```

```{r pca data, echo = FALSE, include=FALSE}
rownames(m2) <- m2$ID
pca_all <- pca(na.omit(d.log2.rb.MC), metadata = m2, removeVar = 0.1)

```

```{r pca plots, fig.height=5, fig.width=5, echo = FALSE, message = FALSE, error=TRUE}
pca_run <- biplot(pca_all,
       showLoadings = F,
       lengthLoadingsArrowsFactor = 1.5,
       x = 'PC1', y = 'PC2',
       sizeLoadingsNames = 5,
       lab = NULL,
       colby = "line", colLegendTitle = "PDX line",
       colkey = col_line,
       pointSize = 3,
       legendPosition = 'right',
       legendLabSize = 16, legendIconSize = 3, legendTitleSize = 16,
       axisLabSize = 16)

pca_line <- biplot(pca_all,
       showLoadings = F,
       lengthLoadingsArrowsFactor = 1.5,
       x = 'PC1', y = 'PC2',
       sizeLoadingsNames = 5,
       lab = NULL,
       colby = "run", colLegendTitle = "MIB run",
       colkey = col_run,
       pointSize = 3,
       legendPosition = 'right',
       legendLabSize = 16, legendIconSize = 3, legendTitleSize = 16,
       axisLabSize = 16)

pca_subtype <- biplot(pca_all,
       showLoadings = F,
       lengthLoadingsArrowsFactor = 1.5,
       x = 'PC1', y = 'PC2',
       sizeLoadingsNames = 5,
       lab = NULL,
       colby = "subtype", colLegendTitle = "subtype",
       colkey = col_SSC,
       pointSize = 3,
       legendPosition = 'right',
       legendLabSize = 16, legendIconSize = 3, legendTitleSize = 16,
       axisLabSize = 16)


png("plots/fig1e_1.png",width=6,height=5,units="in",res=500)
pca_line 
invisible(dev.off())

png("plots/fig1e_2.png",width=6,height=5,units="in",res=500)
pca_run 
invisible(dev.off())

png("plots/fig3b.png",width=6,height=7,units="in",res=500)
pca_subtype
invisible(dev.off())

pca_run
pca_line
pca_subtype

```


```{r consensus cluster data, echo = FALSE, include=FALSE}
title="cc"
cc = ConsensusClusterPlus2(d = as.dist(1-cor(as.matrix(na.omit(d.log2.rb.MC)),method="pearson")),
                                   maxK=12,reps=500,pItem=0.8,pFeature=1,
                                   title=title,clusterAlg="km",distance="euclidean",seed=1, plot="png")

cc15 = ConsensusClusterPlus2(d = as.dist(1-cor(as.matrix(na.omit(d.log2.rb.MC)),method="pearson")),
                                   maxK=15,reps=500,pItem=0.8,pFeature=1,
                                   title=title,clusterAlg="km",distance="euclidean",seed=1, plot="png")

cluster_names <- paste("c", seq(1,12), sep = "")
cluster_names15 <- paste("c", seq(1,15), sep = "")

names(cc) <- cluster_names
names(cc15) <- cluster_names15

cc3_mat <- cc$c3$consensusMatrix
cc10_mat <- cc15$c10$consensusMatrix
cc14_mat <- cc15$c14$consensusMatrix

Colv <- as.dendrogram(cc$c3$consensusTree)
Colv10 <- as.dendrogram(cc15$c10$consensusTree)
Colv14 <- as.dendrogram(cc15$c14$consensusTree)

clust_run <- data.frame(cc15$c10$consensusClass)
clust_run$sample <- rownames(clust_run)
colnames(clust_run) <- c("cluster_run", "ID")
clust_run$cluster_run <- as.character(clust_run$cluster_run)

clust_line <- data.frame(cc15$c14$consensusClass)
clust_line$sample <- rownames(clust_line)
colnames(clust_line) <- c("cluster_line", "ID")
clust_line$cluster_line <- as.character(clust_line$cluster_line)

m_clustcomp <- m2 %>% 
  left_join(clust_line, by = "ID") %>% 
  left_join(clust_run, by = "ID")

adjustedRandIndex(m_clustcomp$line, m_clustcomp$cluster_line)
adjustedRandIndex(m_clustcomp$run, m_clustcomp$cluster_run)

```

```{r consensus clustering plots, fig.height=6, fig.width=7, echo = FALSE, message = FALSE, error=TRUE}
col_ha_cc3 = columnAnnotation(
  cluster = as.matrix(m2$kinotype),
  line = as.matrix(m2$line),
  run = as.matrix(m2$run),
  col = list(cluster = col_kinotypes,
             line = col_line,
             run = col_run),
  simple_anno_size = unit(0.8, "cm"), height = unit(2, "cm"),
  annotation_legend_param = list(cluster = list(title = "cluster", title_gp = gpar(fontsize = 16), 
                                                 labels_gp = gpar(fontsize = 16)),
                                 line = list(title = "line", title_gp = gpar(fontsize = 16), 
                                             labels_gp = gpar(fontsize = 16)),
                                 run = list(title = "run", title_gp = gpar(fontsize = 16), 
                                            labels_gp = gpar(fontsize = 16))),
  annotation_name_gp= gpar(fontsize = 16))

col_ha_line = columnAnnotation(
  show_legend = c(TRUE, FALSE),
  line = m_clustcomp$line,
  k14 = m_clustcomp$cluster_line,
  col = list(line = col_line,
             k14 = col_k14),
  simple_anno_size = unit(0.8, "cm"), height = unit(2, "cm"),
  annotation_legend_param = list(k14 = list(title = "k14", title_gp = gpar(fontsize = 16), 
                                                 labels_gp = gpar(fontsize = 16)),
                                 line = list(title = "line", title_gp = gpar(fontsize = 16), 
                                             labels_gp = gpar(fontsize = 16))),
  annotation_name_gp= gpar(fontsize = 16))


col_ha_run = columnAnnotation(
  show_legend = c(TRUE, FALSE),
  run = m_clustcomp$run,
  k10 = m_clustcomp$cluster_run,
  col = list(run = col_run,
             k10 = col_k10),
  simple_anno_size = unit(0.8, "cm"), height = unit(2, "cm"),
  annotation_legend_param = list(k10 = list(title = "k10", title_gp = gpar(fontsize = 16), 
                                                 labels_gp = gpar(fontsize = 16)),
                                 run = list(title = "run", title_gp = gpar(fontsize = 16), 
                                            labels_gp = gpar(fontsize = 16))),
  annotation_name_gp= gpar(fontsize = 16))


ht_opt(RESET = TRUE)
ht_opt(legend_title_gp = gpar(fontsize = 16), 
       legend_labels_gp = gpar(fontsize = 16)
)

htmp_cc3 <- Heatmap(as.matrix(cc3_mat),
        name = "LFQ",
        col = col_fun,
        cluster_columns = Colv,
        cluster_rows = Colv,
        show_row_names = F,
        column_title = "consensus matrix k=3",
        show_column_names = FALSE,
        column_names_gp = gpar(fontsize = 6),
        top_annotation = col_ha_cc3,
        show_heatmap_legend = FALSE
)

htmp_line <- Heatmap(as.matrix(cc14_mat),
        name = "LFQ",
        col = col_fun,
        cluster_columns = Colv14,
        cluster_rows = Colv14,
        show_row_names = F,
        show_column_names = FALSE,
        column_names_gp = gpar(fontsize = 6),
        top_annotation = col_ha_line,
        column_title = "AdjustedRandIndex = 0.924",
        show_heatmap_legend = FALSE,
)

htmp_run <- Heatmap(as.matrix(cc10_mat),
        name = "LFQ",
        col = col_fun,
        cluster_columns = Colv10,
        cluster_rows = Colv10,
        show_row_names = F,
        show_column_names = FALSE,
        column_names_gp = gpar(fontsize = 6),
        top_annotation = col_ha_run,
        column_title = "AdjustedRandIndex = 0.186",
        show_heatmap_legend = FALSE,
)


png("plots/fig2a.png",width=7,height=6,units="in",res=500)
draw(htmp_cc3, heatmap_legend_side="left", annotation_legend_side="left",
     legend_grouping = "original")
invisible(dev.off())

png("plots/supp1b.png",width=7,height=6,units="in",res=500)
draw(htmp_line, heatmap_legend_side="left", annotation_legend_side="left",
     legend_grouping = "original")
invisible(dev.off())

png("plots/supp1c.png",width=7,height=6,units="in",res=500)
draw(htmp_run, heatmap_legend_side="left", annotation_legend_side="left",
     legend_grouping = "original")
invisible(dev.off())


draw(htmp_cc3, heatmap_legend_side="left", annotation_legend_side="left",
     legend_grouping = "original")

draw(htmp_run, heatmap_legend_side="left", annotation_legend_side="left",
     legend_grouping = "original")

draw(htmp_line, heatmap_legend_side="left", annotation_legend_side="left",
     legend_grouping = "original")

```


```{r limma data, echo = FALSE, include=FALSE}
#kinotypes (3)
sample_k <- d.log2.rb.MC
design_k <- data.frame(k1 = m2$k1 , k2 = m2$k2, k3 = m2$k3)

d_limma_k <- makeContrasts(k1-k2, k1-k3, k3-k2, levels=design_k)
fit_k <- lmFit(sample_k, design_k)

fit_contrast_d_limma_k <- contrasts.fit(fit_k, d_limma_k)
fit_contrast_d_limma_k <- eBayes(fit_contrast_d_limma_k)

top_genes_d_limma_k <- topTable(fit_contrast_d_limma_k, number = 10000, adjust.method = "BH")

featinfo <- YehKinome$df_pdxrna$featInfo
featinfo$ENSG <- gsub("\\..*","", featinfo$ENSG)

featinfo_limma <- featinfo[match(rownames(top_genes_d_limma_k), featinfo$SYMBOL),]
top_genes_d_limma_k$ENSG <- featinfo_limma$ENSG

write.csv(top_genes_d_limma_k, file = "tables/LIMMA_kinotypes_k3.csv")

#subtypes
sample2_all <- dc.log2.rb.MC
design_all <- data.frame(basal = m_line2$basal,classical = m_line2$classical)

d_limma_all <- makeContrasts(RUN = classical-basal, levels=design_all)
fit_all <- lmFit(sample2_all, design_all)

fit_contrast_d_limma_all <- contrasts.fit(fit_all, d_limma_all)
fit_contrast_d_limma_all <- eBayes(fit_contrast_d_limma_all)

top_genes_d_limma_all <- topTable(fit_contrast_d_limma_all, number = 10000, adjust.method = "BH")

featinfo <- YehKinome$df_pdxrna$featInfo
featinfo$ENSG <- gsub("\\..*","", featinfo$ENSG)

featinfo_limma <- featinfo[match(rownames(top_genes_d_limma_all), featinfo$SYMBOL),]
top_genes_d_limma_all$ENSG <- featinfo_limma$ENSG

write.csv(top_genes_d_limma_all, file = "tables/LIMMA_collapsed_all.csv")


```

```{r volcano plots, fig.height=7, fig.width=6, echo = FALSE, message = FALSE, error=TRUE}
k1k2 <- ifelse(top_genes_d_limma_k$k1...k2 < -0.5 & top_genes_d_limma_k$P.Value < 0.05, '#6A3D9A',
  ifelse(top_genes_d_limma_k$k1...k2 > 0.5 & top_genes_d_limma_k$P.Value < 0.05, '#B15928',
         'grey'))
k1k2[is.na(k1k2)] <- 'grey'
names(k1k2)[k1k2 == '#6A3D9A'] <- 'k2'
names(k1k2)[k1k2 == 'grey'] <- 'NS'
names(k1k2)[k1k2 == '#B15928'] <- 'k1'


k1k3 <- ifelse(top_genes_d_limma_k$k1...k3 < -0.5 & top_genes_d_limma_k$P.Value < 0.05, '#f4cc03',
               ifelse(top_genes_d_limma_k$k1...k3 > 0.5 & top_genes_d_limma_k$P.Value < 0.05, '#B15928',
                      'grey'))
k1k3[is.na(k1k3)] <- 'grey'
names(k1k3)[k1k3 == '#f4cc03'] <- 'k3'
names(k1k3)[k1k3 == 'grey'] <- 'NS'
names(k1k3)[k1k3 == '#B15928'] <- 'k1'


k3k2 <- ifelse(top_genes_d_limma_k$k3...k2 < -0.5 & top_genes_d_limma_k$P.Value < 0.05, '#6A3D9A',
               ifelse(top_genes_d_limma_k$k3...k2 > 0.5 & top_genes_d_limma_k$P.Value < 0.05, '#f4cc03',
                      'grey'))
k3k2[is.na(k3k2)] <- 'grey'
names(k3k2)[k3k2 == '#f4cc03'] <- 'k3'
names(k3k2)[k3k2 == 'grey'] <- 'NS'
names(k3k2)[k3k2 == '#6A3D9A'] <- 'k2'



keyvals <- ifelse(
  top_genes_d_limma_all$logFC < -0.5 & top_genes_d_limma_all$P.Value < 0.05, '#fc8803',
  ifelse(top_genes_d_limma_all$logFC > 0.5 & top_genes_d_limma_all$P.Value < 0.05, '#0331fc',
         'grey'))
keyvals[is.na(keyvals)] <- 'grey'
names(keyvals)[keyvals == '#fc8803'] <- 'basal'
names(keyvals)[keyvals == 'grey'] <- 'NS'
names(keyvals)[keyvals == '#0331fc'] <- 'classical'



vplot_k1k2 <- EnhancedVolcano(top_genes_d_limma_k,
                lab = rownames(top_genes_d_limma_k),
                x = "k1...k2",
                y = "P.Value",
                title = "",
                subtitle = "",
                caption = "",
                gridlines.major = TRUE, gridlines.minor = TRUE,
                legendPosition = "top",
                selectLab = rownames(top_genes_d_limma_k)[which(names(k1k2) %in% c('k1', 'k2'))],
                pCutoff = 5e-2,
                colCustom = k1k2,
                drawConnectors = T,
                FCcutoff = 0.5,
                pointSize = 2,
                legendLabels=c('Not sig.','Log2FC','adj.p-value',
                               'adj.p-value & Log2FC'),
                labSize = 4) + ylab(bquote(-Log[10]~ P.Value)) + ylim(0, 22)  +  font("xlab", size = 14, face = "bold") + font("ylab", size = 14, face = "bold") + font("xy.text", color = "black", size = 14)

vplot_k1k3 <- EnhancedVolcano(top_genes_d_limma_k,
                lab = rownames(top_genes_d_limma_k),
                x = "k1...k3",
                y = "P.Value",
                title = "",
                subtitle = "",
                caption = "",                
                gridlines.major = TRUE, gridlines.minor = TRUE,
                legendPosition = "top",
                selectLab = rownames(top_genes_d_limma_k)[which(names(k1k3) %in% c('k1', 'k3'))],
                pCutoff = 5e-2,
                colCustom = k1k3,
                drawConnectors = T,
                FCcutoff = 0.5,
                pointSize = 2,
                legendLabels=c('Not sig.','Log2FC','adj.p-value',
                               'adj.p-value & Log2FC'),
                labSize = 4) + ylab(bquote(-Log[10]~ P.Value))+ ylim(0, 22)  +  font("xlab", size = 14, face = "bold") + font("ylab", size = 14, face = "bold") + font("xy.text", color = "black", size = 14)

vplot_k3k2 <- EnhancedVolcano(top_genes_d_limma_k,
                lab = rownames(top_genes_d_limma_k),
                x = "k3...k2",
                y = "P.Value",
                title = "",
                subtitle = "",
                caption = "",                
                gridlines.major = TRUE, gridlines.minor = TRUE,
                legendPosition = "top",
                selectLab = rownames(top_genes_d_limma_k)[which(names(k3k2) %in% c('k3', 'k2'))],
                pCutoff = 5e-2,
                colCustom = k3k2,
                drawConnectors = T,
                FCcutoff = 0.5,
                pointSize = 2,
                legendLabels=c('Not sig.','Log2FC','adj.p-value',
                               'adj.p-value & Log2FC'),
                labSize = 4) + ylab(bquote(-Log[10]~ P.Value))+ ylim(0, 22)  +  font("xlab", size = 14, face = "bold") + font("ylab", size = 14, face = "bold") + font("xy.text", color = "black", size = 14)



vplot_ssc <- EnhancedVolcano(top_genes_d_limma_all,
                lab = rownames(top_genes_d_limma_all),
                x = "logFC",
                y = "P.Value",
                title = "",
                subtitle = "",
                caption = "",                
                gridlines.major = TRUE, gridlines.minor = TRUE,
                legendPosition = "top",
                selectLab = rownames(top_genes_d_limma_all)[which(names(keyvals) %in% c('basal', 'classical'))],
                pCutoff = 5e-2,
                colCustom = keyvals,
                FCcutoff = 0.5,
                pointSize = 3,
                legendLabels=c('Not sig.','Log2FC','adj.p-value',
                               'adj.p-value & Log2FC'),
                labSize = 4)+ ylim(0, 4.5) + xlim(-3, 3) + ylab(bquote(-Log[10]~ P.Value)) +  font("xlab", size = 14, face = "bold") + font("ylab", size = 14, face = "bold") + font("xy.text", color = "black", size = 14)

png("plots/fig2b_1.png",width=3.5,height=7,units="in",res=500)
vplot_k1k2
invisible(dev.off())

png("plots/fig2b_2.png",width=3.5,height=7,units="in",res=500)
vplot_k1k3
invisible(dev.off())

png("plots/fig2b_3.png",width=3.5,height=7,units="in",res=500)
vplot_k3k2
invisible(dev.off())

png("plots/fig3d.png",width=6,height=7,units="in",res=1000)
vplot_ssc
invisible(dev.off())

vplot_k1k2
vplot_k1k3
vplot_k3k2
vplot_ssc


```






```{r GO analysis data, echo = FALSE, include=FALSE}
#kinotypes
topgenes_kinotype <- YehKinome$df_limma_k3

k1_genes <- data.frame(mapIds(org.Hs.eg.db, keys = topgenes_kinotype$k1$kinase,
                              column = "ENTREZID", keytype = "SYMBOL"))
colnames(k1_genes) <- "gene"

k2_genes <- data.frame(mapIds(org.Hs.eg.db, keys = topgenes_kinotype$k2$kinase,
                              column = "ENTREZID", keytype = "SYMBOL"))
colnames(k2_genes) <- "gene"

k3_genes <- data.frame(mapIds(org.Hs.eg.db, keys = topgenes_kinotype$k3$kinase,
                              column = "ENTREZID", keytype = "SYMBOL"))
colnames(k3_genes) <- "gene"

GO_comp_k <- list(k1 = k1_genes$gene, 
                k2 = k2_genes$gene,
                k3 = k3_genes$gene)

kegg_comp_k <- compareCluster(geneCluster = GO_comp_k, fun = enrichKEGG,
                            pvalueCutoff = 0.05)

#subtypes
pval_genes_basal <- na.omit(top_genes_d_limma_all[top_genes_d_limma_all[,4]<0.05 & top_genes_d_limma_all[,1] < 0.5,])
pval_genes_classical <- na.omit(top_genes_d_limma_all[top_genes_d_limma_all[,4]<0.05 & top_genes_d_limma_all[,1] > 0.5,])

GO_genes_basal <- data.frame(mapIds(org.Hs.eg.db, keys = rownames(pval_genes_basal),
                              column = "ENTREZID", keytype = "SYMBOL"))
colnames(GO_genes_basal) <- "gene"


GO_genes_classical <- data.frame(mapIds(org.Hs.eg.db, keys = rownames(pval_genes_classical),
                                    column = "ENTREZID", keytype = "SYMBOL"))
colnames(GO_genes_classical) <- "gene"

GO_comp <- list(basal = GO_genes_basal$gene, 
                classical = GO_genes_classical$gene)

kegg_comp <- compareCluster(geneCluster = GO_comp, fun = enrichKEGG,
                            pvalueCutoff = 0.05)

kegg_comp@compareClusterResult$Description <- strtrim(kegg_comp@compareClusterResult$Description, 45)

```


```{r GO analysis plots, fig.height=8, fig.width=6, echo = FALSE, message = FALSE, error=TRUE}
k3_kegg <- dotplot(kegg_comp_k, showCategory = 10, includeAll = F) + xlab("subtype") + font("x.text", size = 16)  + scale_x_discrete(labels= c("k1", "k2", "k3"))

k2_kegg <- dotplot(kegg_comp, showCategory = 15, includeAll = F) + xlab("subtype") + font("x.text", size = 14)  + scale_x_discrete(labels= c("basal", "classical"))

png("plots/fig2c.png",width=6,height=8,units="in",res=500)
k3_kegg
invisible(dev.off())

png("plots/fig3e.png",width=6,height=8,units="in",res=500)
k2_kegg
invisible(dev.off())

k3_kegg
k2_kegg
```

```{r boxplots data, echo = FALSE, include=FALSE}

#define top genes kinotypes
pval_genes_k <- na.omit(top_genes_d_limma_k[top_genes_d_limma_k[,7]<0.05,])

pval_k1 <- d.log2[match(rownames(k1_genes), rownames(d.log2)),1:101]
pval_k2 <- d.log2[match(rownames(k2_genes), rownames(d.log2)),1:101]
pval_k3 <- d.log2[match(rownames(k3_genes), rownames(d.log2)),1:101]

pval_k1 <- pval_k1 %>% t() %>% as.data.frame() %>% 
  mutate(sample = m2$ID, .before = 1) %>% 
  mutate(kinotype = m2$kinotype, .before = 1)
pval_k2 <- pval_k2 %>% t() %>% as.data.frame() %>% 
  mutate(sample = m2$ID, .before = 1) %>% 
  mutate(kinotype = m2$kinotype, .before = 1)
pval_k3 <- pval_k3 %>% t() %>% as.data.frame() %>% 
  mutate(sample = m2$ID, .before = 1) %>% 
  mutate(kinotype = m2$kinotype, .before = 1)

pval_k1_t5 <- pval_k1 %>% dplyr::select(-c('MAPK13', 'SYK', 'NAGK', 'HK2'))
pval_k2_t5 <- pval_k2 %>% dplyr::select(-c('PRKG1', 'EPHB6', 'MARK1', 'NUAK1'))
pval_k3_t5 <- pval_k3

k1_genes$kinase <- rownames(k1_genes)
k2_genes$kinase <- rownames(k2_genes)
k3_genes$kinase <- rownames(k3_genes)

k1_genes_t5 <- k1_genes[!(row.names(k1_genes) %in% c('MAPK13', 'SYK', 'NAGK', 'HK2')),]
k2_genes_t5 <- k2_genes[!(row.names(k2_genes) %in% c('PRKG1', 'EPHB6', 'MARK1', 'NUAK1')),]
k3_genes_t5 <- k3_genes

my_comparisons <- list( c("k1", "k2"), c("k2", "k3"), c("k1", "k3") )

#define top genes subtypes ttest
top_ssc <- data.frame(t(d.log2[,1:101])) %>% 
  mutate(subtype = m2$subtype) %>%
  pivot_longer(-subtype, names_to = "variables", values_to = "value") %>% 
  na.omit() %>% 
  group_by(variables) %>% 
  filter(all(c("basal", "classical") %in% subtype))

top_ssc_ttest <- top_ssc %>%
  group_by(variables) %>%
  t_test(value ~ subtype, detailed = T) %>%
  adjust_pvalue(method = "BH") %>%
  add_significance()

write.csv(top_ssc_ttest, file = "tables/t-test_subtypes.csv")

top_basal <- top_ssc_ttest %>% 
  filter(estimate > 0.7, p < 0.05) %>% 
  arrange(p.adj) %>% 
  pull(variables)

top_classical <- top_ssc_ttest %>% 
  filter(estimate < -0.4, p < 0.05) %>% 
  arrange(p.adj) %>% 
  pull(variables)

pval_basal <- d.log2 %>%
  slice(match(top_basal, rownames(d.log2))) %>%
  select(1:101) %>%                                           
  t() %>%                                                     
  as.data.frame() %>%                                         
  rownames_to_column("sample") %>%                            
  mutate(subtype = m2$subtype)

pval_classical <- d.log2 %>%
  slice(match(top_classical, rownames(d.log2))) %>%
  select(1:101) %>%                                           
  t() %>%                                                     
  as.data.frame() %>%                                         
  rownames_to_column("sample") %>%                            
  mutate(subtype = m2$subtype)

#RTKs
RTK_list <- c("CSF1R", "EPHA2", "EPHB2", "EPHB6", "EGFR", "IGF1R", "INSR", "MET")
RTK_df <- d.log2[,1:101] %>% 
  t() %>% 
  as.data.frame() %>% 
  rownames_to_column("sample") %>% 
  mutate(subtype = m2$subtype,
         line = m2$line) %>% 
  select(subtype, line, all_of(RTK_list))

```


```{r kinotype boxplots plots, fig.height=2, fig.width=6, echo = FALSE, message = FALSE, error=TRUE}
#kinotypes boxplots
my_plot_list = list()
for(i in 1:5){
  p <- ggboxplot(pval_k1_t5, x = "kinotype", y = rownames(k1_genes_t5)[i], 
                 color = "kinotype", palette = c("#B15928","#6A3D9A","#f4cc03"), legend = "none", title = rownames(k1_genes_t5)[i],
                 font.main = c(10, "bold", "black"),
                 font.x = 8,
                 font.ytickslab = 8,
                 font.xtickslab = 8,
                 ylab = FALSE,
                 add = "jitter",
                 add.params = list(size = 0.5)) + stat_compare_means(comparisons = my_comparisons, size = 2.4) + scale_y_continuous(expand = expansion(mult = c(0, 0.1)))
  my_plot_list[[i]] <- p
}

k1_boxplots_t5 <- ggarrange(plotlist = my_plot_list, nrow = 1, ncol = 5)
annotate_figure(k1_boxplots_t5, left = text_grob("Log2(LFQ)", color = "black",face = "bold", rot = 90, size = 8))

my_plot_list = list()
for(i in 1:5){
  p <- ggboxplot(pval_k2_t5, x = "kinotype", y = rownames(k2_genes_t5)[i], 
                 color = "kinotype", palette = c("#B15928","#6A3D9A","#f4cc03"), legend = "none", title = rownames(k2_genes_t5)[i],
                 font.main = c(10, "bold", "black"),
                 font.x = 8,
                 font.ytickslab = 8,
                 font.xtickslab = 8,
                 ylab = FALSE,
                 add = "jitter",
                 add.params = list(size = 0.5)) + stat_compare_means(comparisons = my_comparisons, size = 2.4) + scale_y_continuous(expand = expansion(mult = c(0, 0.1)))
  my_plot_list[[i]] <- p
}
k2_boxplots_t5 <- ggarrange(plotlist = my_plot_list, nrow = 1, ncol = 5)
annotate_figure(k2_boxplots_t5, left = text_grob("Log2(LFQ)", color = "black",face = "bold", rot = 90, size = 8))

my_plot_list = list()
for(i in 1:5){
  p <- ggboxplot(pval_k3_t5, x = "kinotype", y = rownames(k3_genes_t5)[i], 
                 color = "kinotype", palette = c("#B15928","#6A3D9A","#f4cc03"), legend = "none", title = rownames(k3_genes_t5)[i],
                 font.main = c(10, "bold", "black"),
                 font.x = 8,
                 font.ytickslab = 8,
                 font.xtickslab = 8,
                 ylab = FALSE,
                 add = "jitter",
                 add.params = list(size = 0.5)) + stat_compare_means(comparisons = my_comparisons, size = 2.4) + scale_y_continuous(expand = expansion(mult = c(0, 0.1)))
  my_plot_list[[i]] <- p
}

k3_boxplots_t5 <- ggarrange(plotlist = my_plot_list, nrow = 1, ncol = 5)
annotate_figure(k3_boxplots_t5, left = text_grob("Log2(LFQ)", color = "black",face = "bold", rot = 90, size = 8))

png("plots/fig2d_1.png",width=6,height=1.7,units="in",res=500)
k1_boxplots_t5
invisible(dev.off())

png("plots/fig2d_2.png",width=6,height=1.7,units="in",res=500)
k2_boxplots_t5
invisible(dev.off())

png("plots/fig2d_3.png",width=6,height=1.7,units="in",res=500)
k3_boxplots_t5
invisible(dev.off())
```

```{r subtype boxplots plots, fig.height=9, fig.width=9, echo = FALSE, message = FALSE, error=TRUE}
#subtypes boxplots
my_plot_list = list()
for(i in 1:20){
  p <- ggboxplot(pval_classical, x = "subtype", y = top_classical[i], 
                 color = "subtype", palette = c("#0331fc","#fc8803"), legend = "none", title = top_classical[i],
                 font.main = c(12, "bold", "black"),
                 font.x = 12,
                 font.xtickslab = 12,
                 ylab = FALSE,
                 add = "jitter") + stat_compare_means(size = 5, aes(label = paste0("p =", ..p.format..)),
                                                      method = "t.test",
                 ) +  scale_y_continuous(expand = expansion(mult = c(0, 0.1)))
  my_plot_list[[i]] <- p
}
classical_boxplots <- ggarrange(plotlist = my_plot_list, nrow = 4, ncol =5)
annotate_figure(classical_boxplots, left = text_grob("Log2(LFQ)", color = "black",face = "bold", rot = 90))

my_plot_list = list()
for(i in 1:20){
  p <- ggboxplot(pval_basal, x = "subtype", y = top_basal[i], 
                 color = "subtype", palette = c("#0331fc","#fc8803"), legend = "none", title = top_basal[i],
                 font.main = c(12, "bold", "black"),
                 font.x = 12,
                 font.xtickslab = 12,
                 ylab = FALSE,
                 add = "jitter") + stat_compare_means(size = 5, aes(label = paste0("p =", ..p.format..)),
                                                      method = "t.test",
                 ) +  scale_y_continuous(expand = expansion(mult = c(0, 0.1)))
  my_plot_list[[i]] <- p
}
basal_boxplots <- ggarrange(plotlist = my_plot_list, nrow = 4, ncol =5)
annotate_figure(basal_boxplots, left = text_grob("Log2(LFQ)", color = "black",face = "bold", rot = 90))

png("plots/fig4a.png",width=11,height=10,units="in",res=500)
classical_boxplots
invisible(dev.off())

png("plots/fig4b.png",width=11,height=10,units="in",res=500)
basal_boxplots
invisible(dev.off())

#RTKs
my_plot_list = list()
for(i in 1:8){
  p <- ggboxplot(RTK_df, x = "subtype", y = RTK_list[i], 
                 color = "subtype", palette = c("#0331fc","#fc8803"), legend = "none", title = RTK_list[i],
                 font.main = c(12, "bold", "black"),
                 font.x = 12,
                 font.xtickslab = 12,
                 ylab = FALSE,
                 add = "jitter") + stat_compare_means(size = 5, aes(label = paste0("p = ", ..p.format..)),
                                                      method = "t.test",
                 ) +  scale_y_continuous(expand = expansion(mult = c(0, 0.1)))
  my_plot_list[[i]] <- p
}

RTK_boxplots <- ggarrange(plotlist = my_plot_list, nrow = 2, ncol = 4)
annotate_figure(RTK_boxplots, left = text_grob("Log2(LFQ)", color = "black",face = "bold", rot = 90))


png("plots/supp4a.png",width=7,height=5,units="in",res=500)
RTK_boxplots
invisible(dev.off())
```

```{r protein heatmap data, echo = FALSE, include=FALSE}
d_heatmap <- na.omit(d.log2.rb.MC)
d_heatmap_scaled <- data.frame(t(scale(t(d_heatmap))))

col_ha = columnAnnotation(
  KinaseCluster = as.matrix(m2$kinotype),
  subtype = as.matrix(m2$subtype),
  line = as.matrix(m2$line),
  run = as.matrix(m2$run),
  col = list(subtype = col_SSC,
             KinaseCluster = col_kinotypes,
             line = col_line,
             run = col_run),
  simple_anno_size = unit(0.8, "cm"), height = unit(2, "cm"),
  annotation_legend_param = list(subtype = list(title = "subtype", title_gp = gpar(fontsize = 16), 
                                            labels_gp = gpar(fontsize = 16)),
                                 KinaseCluster = list(title = "KinaseCluster", title_gp = gpar(fontsize = 16), 
                                                 labels_gp = gpar(fontsize = 16)),
                                 line = list(title = "line", title_gp = gpar(fontsize = 16), 
                                             labels_gp = gpar(fontsize = 16)),
                                 run = list(title = "run", title_gp = gpar(fontsize = 16), 
                                            labels_gp = gpar(fontsize = 16))),
  annotation_name_gp= gpar(fontsize = 16))

datCol <- as.dist(1-cor(as.matrix(d_heatmap),method="pearson"))
datRow <- as.dist(1-cor(t(as.matrix(d_heatmap)),method="pearson"))

Colv <- as.dendrogram(
  ConsensusClusterPlus::ConsensusClusterPlus(
    d = datCol,
    seed = 1, pFeature = 1, pItem = 0.8,
    maxK = 3, reps=500, distance="euclidean",
    clusterAlg="km")[[2]]$consensusTree)

Rowv <- as.dendrogram(
  ConsensusClusterPlus::ConsensusClusterPlus(
    d = datRow,
    seed = 1, pFeature = 1, pItem = 0.8,
    maxK = 3, reps=500, distance="euclidean",
    clusterAlg="km")[[2]]$consensusTree)

```

```{r protein heatmap plots, fig.height=8, fig.width=12, echo = FALSE, message = FALSE, error=TRUE}
ht_opt(RESET = TRUE)
ht_opt(legend_title_gp = gpar(fontsize = 16), 
       legend_labels_gp = gpar(fontsize = 16))

htmap_k2 <- Heatmap(as.matrix(d_heatmap_scaled),
        name = "LFQ",
        col = col_fun_htmp,
        cluster_columns = Colv,
        cluster_rows = Rowv,
        show_row_names = F,
        show_column_names = FALSE,
        column_names_gp = gpar(fontsize = 6),
        top_annotation = col_ha,
        heatmap_legend_param = list(
          at = c(-4, 0, 4),
          title = "LFQ",
          legend_height = unit(4, "cm"),
          title_position = "topcenter")
)

draw(htmap_k2, heatmap_legend_side="left", annotation_legend_side="left",
     legend_grouping = "original")

png("plots/fig3a.png",width=12,height=8,units="in",res=500)
draw(htmap_k2, heatmap_legend_side="left", annotation_legend_side="left",
     legend_grouping = "original")
invisible(dev.off())
```

```{r correlation across lines data, echo = FALSE, include=FALSE}
fn_corr_line_run <- function(x){
  val <- m %>% 
    filter(line == x) %>% 
    pull(run) %>% 
    unique()
  
  corr_m <- m %>%
    filter(run %in% val) %>% 
    filter(!grepl("pool", line)) %>% 
    mutate(line = ifelse(line !=x, "other", line))
  
  corr_m$line <- factor(corr_m$line, levels = c(x, "other"))
  df_corr <- na.omit(d.log2.rb.MC[,colnames(d.log2.rb.MC) %in% corr_m$ID])
  corr_df <- cor(df_corr)
  
  col_ha = columnAnnotation(
    line = corr_m$line,
    run = corr_m$run,
    col = list(line = c("P319T1" = "#D62728",
                        "P710T1" = "#D62728",
                        "P203T1" = "#D62728",
                        "P616T1" = "#D62728",
                        "P109T1" = "#D62728",
                        "P411T1" = "#D62728",
                        "P125T2" = "#D62728",
                        "P119T1" = "#D62728",
                        "P225T1" = "#D62728",
                        "PancT6" = "#D62728",
                        "P508T1" = "#D62728",
                        "P902T1B" = "#D62728",
                        "P806T1" = "#D62728",
                        "P713T1" = "#D62728",
                        "other" = "#7F7F7F"),
               run = col_run))
  
  Heatmap(as.matrix(corr_df),
          column_title = x,
          cluster_columns = T,
          cluster_rows = T,
          show_row_names = FALSE,
          show_column_names = FALSE,
          column_names_gp = gpar(fontsize = 2),
          top_annotation = col_ha,
          heatmap_legend_param = list(
            at = c(-1, 0, 1),
            title = "corr",
            legend_height = unit(4, "cm"),
            title_position = "topcenter"))
}

```

```{r correlation across lines plots, fig.height=4, fig.width=3.5, echo = FALSE, message = FALSE, error=TRUE}
corrplots_line_run <- lapply(unique(m2$line), fn_corr_line_run)
names(corrplots_line_run) <- unique(m2$line)

for (i in seq_along(corrplots_line_run)) {
  plot_name <- names(corrplots_line_run)[i]
  png(filename = file.path("plots", paste0("supp1a_", plot_name, ".png")), 
      width = 3.5, height = 4, units = "in",
      res = 500)
  print(corrplots_line_run[[i]])
  invisible(dev.off())
}

corrplots_line_run
```

```{r RNAseq graphs data, echo = FALSE, include=FALSE}
RNA_meta_MIB <- m_rna
RNA_meta_MIB2 <- m_rna

RNA_ex_MIB <- RNA_ex[,match(RNA_meta_MIB$ID, names(RNA_ex))]
RNA_ex_ENSG_MIB <- RNA_ex_ENSG[,match(RNA_meta_MIB$ID, names(RNA_ex_ENSG))]

RNA_ex_MIB_TSP <- RNA_ex_MIB[match(TSP_geneset$gene, rownames(RNA_ex_MIB)),]
RNA_ex_MIB_TSP_scaled <- data.frame(t(scale(t(as.matrix(RNA_ex_MIB_TSP)))))

RNA_ex_MIB_collisson <- RNA_ex_MIB[match(collisson_geneset$gene, rownames(RNA_ex_MIB)),]
RNA_ex_MIB_collisson_scaled <- data.frame(t(scale(t(as.matrix(RNA_ex_MIB_collisson)))))

RNA_ex_MIB_moffitt <- RNA_ex_MIB[match(moffitt_geneset$gene, rownames(RNA_ex_MIB)),]
RNA_ex_MIB_moffitt <- na.omit(RNA_ex_MIB_moffitt)
RNA_ex_MIB_moffitt_scaled <- data.frame(t(scale(t(as.matrix(RNA_ex_MIB_moffitt)))))
RNA_ex_MIB_moffitt_scaled <- na.omit(RNA_ex_MIB_moffitt_scaled)

RNA_ex_MIB_bailey <- RNA_ex_ENSG_MIB[match(bailey_geneset$gene, rownames(RNA_ex_ENSG_MIB)),]
RNA_ex_MIB_bailey <- na.omit(RNA_ex_MIB_bailey)
RNA_ex_MIB_bailey_scaled <- data.frame(t(scale(t(as.matrix(RNA_ex_MIB_bailey)))))
RNA_ex_MIB_bailey_scaled <- na.omit(RNA_ex_MIB_bailey_scaled)

datCol_TSP <- as.dist(1-cor(as.matrix(RNA_ex_MIB_TSP),method="pearson"))
datRow_TSP <- as.dist(1-cor(t(as.matrix(RNA_ex_MIB_TSP)),method="pearson"))

datCol_collisson <- as.dist(1-cor(as.matrix(RNA_ex_MIB_collisson),method="pearson"))
datRow_collisson <- as.dist(1-cor(t(as.matrix(RNA_ex_MIB_collisson)),method="pearson"))

#because there are missing values in Moffitt/Bailey, need to match clustering data to scaled data that have na.omit
datCol_moffitt <- as.dist(1-cor(as.matrix(RNA_ex_MIB_moffitt[match(rownames(RNA_ex_MIB_moffitt_scaled), rownames(RNA_ex_MIB_moffitt)),]),method="pearson"))
datRow_moffitt <- as.dist(1-cor(t(as.matrix(RNA_ex_MIB_moffitt[match(rownames(RNA_ex_MIB_moffitt_scaled), rownames(RNA_ex_MIB_moffitt)),])),method="pearson"))

datCol_bailey <- as.dist(1-cor(as.matrix(RNA_ex_MIB_bailey[match(rownames(RNA_ex_MIB_bailey_scaled), rownames(RNA_ex_MIB_bailey)),]),method="pearson"))
datRow_bailey <- as.dist(1-cor(t(as.matrix(RNA_ex_MIB_bailey[match(rownames(RNA_ex_MIB_bailey_scaled), rownames(RNA_ex_MIB_bailey)),])),method="pearson"))

col_ha1 = columnAnnotation(
  #SSC = ca4,
  line = as.matrix(RNA_meta_MIB2$line),
  col = list(#SSC = c("classical" = "#0331fc", "basal" = "#fc8803"),
    line = col_line),
  simple_anno_size = unit(1, "cm"), height = unit(4, "cm"),
  annotation_legend_param = list(#SSC = list(title = "subtype", title_gp = gpar(fontsize = 18), 
    #labels_gp = gpar(fontsize = 18)),
    line = list(title = "line", title_gp = gpar(fontsize = 14), 
                labels_gp = gpar(fontsize = 14))),
  annotation_name_gp= gpar(fontsize = 14))


col_ha2 = columnAnnotation(
  SSC = as.matrix(RNA_meta_MIB2$PurIST),
  line = as.matrix(RNA_meta_MIB2$line),
  col = list(SSC = col_SSC,
             line = col_line),
  simple_anno_size = unit(1, "cm"), height = unit(4, "cm"),
  annotation_legend_param = list(SSC = list(title = "subtype", title_gp = gpar(fontsize = 12), 
                                            labels_gp = gpar(fontsize = 12)),
                                 line = list(title = "line", title_gp = gpar(fontsize = 12), 
                                             labels_gp = gpar(fontsize = 12))),
  annotation_name_gp= gpar(fontsize = 12))

ra_TSP <- as.matrix(TSP_geneset$subtype)
ra_collisson <- as.matrix(collisson_geneset$subtype)
ra_moffitt <- as.matrix(moffitt_geneset[match(rownames(RNA_ex_MIB_moffitt_scaled), moffitt_geneset$gene),]$subtype)
ra_bailey <- as.matrix(bailey_geneset[match(rownames(RNA_ex_MIB_bailey_scaled), bailey_geneset$gene),]$subtype)


row_TSP = rowAnnotation(
  subtype = ra_TSP,
  show_legend = T,
  show_annotation_name = F,
  annotation_name_gp= gpar(fontsize = 14),
  simple_anno_size = unit(0.5, "cm"),
  col = list(subtype = c("classical" = "#0331fc", "basal" = "#fc8803")),
  annotation_legend_param = list(subtype = list(title = "subtype", title_gp = gpar(fontsize = 14), 
                                                labels_gp = gpar(fontsize = 14))))

row_collison = rowAnnotation(
  subtype = ra_collisson,
  show_legend = T,
  show_annotation_name = F,
  annotation_name_gp= gpar(fontsize = 18),
  simple_anno_size = unit(0.8, "cm"),
  col = list(subtype = c("classical" = "purple", "exocrine" = "yellow", "QM" = "blue")),
  annotation_legend_param = list(subtype = list(title = "subtype", title_gp = gpar(fontsize = 18), 
                                                labels_gp = gpar(fontsize = 18))))

row_moffitt = rowAnnotation(
  subtype = ra_moffitt,
  show_legend = T,
  show_annotation_name = F,
  annotation_name_gp= gpar(fontsize = 18),
  simple_anno_size = unit(0.8, "cm"),
  col = list(subtype = c("classical" = "#0331fc", "basal" = "#fc8803")),
  annotation_legend_param = list(subtype = list(title = "subtype", title_gp = gpar(fontsize = 18), 
                                                labels_gp = gpar(fontsize = 18))))
row_bailey = rowAnnotation(
  subtype = ra_bailey,
  show_legend = T,
  show_annotation_name = F,
  annotation_name_gp= gpar(fontsize = 18),
  simple_anno_size = unit(0.8, "cm"),
  col = list(subtype = c("squamous" = "#518aa7", "ADEX" = "#674529", "progenitor" = "#f4c93e", "immunogenic" = "#ba242d")),
  annotation_legend_param = list(subtype = list(title = "subtype", title_gp = gpar(fontsize = 18), 
                                                labels_gp = gpar(fontsize = 18))))

Colv_TSP <- as.dendrogram(
  ConsensusClusterPlus::ConsensusClusterPlus(
    d = datCol_TSP,
    seed = 1, pFeature = 1, pItem = 0.8,
    maxK = 3, reps=500, distance="euclidean",
    clusterAlg="km")[[2]]$consensusTree)

Rowv_TSP <- as.dendrogram(
  ConsensusClusterPlus::ConsensusClusterPlus(
    d = datRow_TSP,
    seed = 1, pFeature = 1, pItem = 0.8,
    maxK = 3, reps=500, distance="euclidean",
    clusterAlg="km")[[2]]$consensusTree)

Colv_collisson <- as.dendrogram(
  ConsensusClusterPlus::ConsensusClusterPlus(
    d = datCol_collisson,
    seed = 1, pFeature = 1, pItem = 0.8,
    maxK = 4, reps=500, distance="euclidean",
    clusterAlg="km")[[3]]$consensusTree)

Rowv_collisson <- as.dendrogram(
  ConsensusClusterPlus::ConsensusClusterPlus(
    d = datRow_collisson,
    seed = 1, pFeature = 1, pItem = 0.8,
    maxK = 4, reps=500, distance="euclidean",
    clusterAlg="km")[[3]]$consensusTree)

Colv_moffitt <- as.dendrogram(
  ConsensusClusterPlus::ConsensusClusterPlus(
    d = datCol_moffitt,
    seed = 1, pFeature = 1, pItem = 0.8,
    maxK = 3, reps=500, distance="euclidean",
    clusterAlg="km")[[2]]$consensusTree)

Rowv_moffitt <- as.dendrogram(
  ConsensusClusterPlus::ConsensusClusterPlus(
    d = datRow_moffitt,
    seed = 1, pFeature = 1, pItem = 0.8,
    maxK = 3, reps=500, distance="euclidean",
    clusterAlg="km")[[2]]$consensusTree)

Colv_bailey <- as.dendrogram(
  ConsensusClusterPlus::ConsensusClusterPlus(
    d = datCol_bailey,
    seed = 1, pFeature = 1, pItem = 0.8,
    maxK = 5, reps=500, distance="euclidean",
    clusterAlg="km")[[4]]$consensusTree)

Rowv_bailey <- as.dendrogram(
  ConsensusClusterPlus::ConsensusClusterPlus(
    d = datRow_bailey,
    seed = 1, pFeature = 1, pItem = 0.8,
    maxK = 5, reps=500, distance="euclidean",
    clusterAlg="km")[[4]]$consensusTree)

```

```{r RNAseq graphs plots, fig.height=8, fig.width=9, echo = FALSE, message = FALSE, error=TRUE}
ht_opt(RESET = TRUE)
ht_opt(legend_title_gp = gpar(fontsize = 14), 
       legend_labels_gp = gpar(fontsize = 14)
)

TSP_hm <- Heatmap(as.matrix(RNA_ex_MIB_TSP_scaled),
        name = "LFQ",
        cluster_columns = Colv_TSP,
        cluster_rows = F,
        show_row_names = TRUE,
        row_names_side = "left",
        show_column_names = FALSE,
        column_names_gp = gpar(fontsize = 2),
        row_names_gp = gpar(fontsize = 8),
        top_annotation = col_ha1,
        left_annotation = row_TSP,
        heatmap_legend_param = list(
          at = c(-4, 0, 4),
          title = "exprs",
          legend_height = unit(2, "cm"),
          title_position = "topcenter")
)

draw(TSP_hm, heatmap_legend_side="left", annotation_legend_side="left",
     legend_grouping = "original")

png("plots/supp3a.png",width=6,height=4,units="in",res=500)
draw(TSP_hm, heatmap_legend_side="left", annotation_legend_side="left",
     legend_grouping = "original")
invisible(dev.off())


ht_opt(RESET = TRUE)
ht_opt(legend_title_gp = gpar(fontsize = 16), 
       legend_labels_gp = gpar(fontsize = 16)
)

collison_hm <- Heatmap(as.matrix(RNA_ex_MIB_collisson_scaled),
        name = "LFQ",
        cluster_columns = Colv_collisson,
        cluster_rows = F,
        show_row_names = F,
        row_names_side = "left",
        show_column_names = FALSE,
        column_names_gp = gpar(fontsize = 6),
        row_names_gp = gpar(fontsize = 6.5),
        top_annotation = col_ha2,
        left_annotation = row_collison,
        heatmap_legend_param = list(
          at = c(-4, 0, 4),
          title = "exprs",
          legend_height = unit(3, "cm"),
          title_position = "topcenter")
)

moffitt_hm <- Heatmap(as.matrix(RNA_ex_MIB_moffitt_scaled),
        name = "LFQ",
        cluster_columns = Colv_moffitt,
        cluster_rows = F,
        show_row_names = F,
        row_names_side = "left",
        show_column_names = FALSE,
        column_names_gp = gpar(fontsize = 6),
        row_names_gp = gpar(fontsize = 6.5),
        top_annotation = col_ha2,
        left_annotation = row_moffitt,
        heatmap_legend_param = list(
          at = c(-4, 0, 4),
          title = "exprs",
          legend_height = unit(3, "cm"),
          title_position = "topcenter")
)

bailey_hm <- Heatmap(as.matrix(RNA_ex_MIB_bailey_scaled),
        name = "LFQ",
        cluster_columns = Colv_bailey,
        cluster_rows = F,
        show_row_names = F,
        row_names_side = "left",
        show_column_names = FALSE,
        column_names_gp = gpar(fontsize = 6),
        row_names_gp = gpar(fontsize = 6.5),
        top_annotation = col_ha2,
        left_annotation = row_bailey,
        heatmap_legend_param = list(
          at = c(-4, 0, 4),
          title = "exprs",
          legend_height = unit(3, "cm"),
          title_position = "topcenter")
)

png("plots/supp3b.png",width=9,height=8,units="in",res=500)
moffitt_hm
invisible(dev.off())

png("plots/supp3c.png",width=9,height=8,units="in",res=500)
collison_hm
invisible(dev.off())

png("plots/supp3d.png",width=9,height=8,units="in",res=500)
bailey_hm
invisible(dev.off())

moffitt_hm
collison_hm
bailey_hm
```



```{r RNA-prot correlation data, echo = FALSE, include=FALSE}
m_prot <- m2[match(RNA_meta_MIB$ID, m2$ID),]

#collapsed protein meta
m_prot_line <- m_prot
m_prot_line$ID <- sub("_[^_]+$", "", m_prot$ID)

ID_order <- unique(m_prot_line$ID)

concat_unique <- function(x){paste(unique(x),  collapse=',')}

m_prot_line <- m_prot_line %>% 
  group_by(ID) %>% 
  summarise_all(concat_unique)
m_prot_line <- m_prot_line[match(ID_order, m_prot_line$ID),]

#collapsed RNA meta
m_RNA_line <- RNA_meta_MIB2

ID_order <- unique(m_RNA_line$line) #define order of samples

m_RNA_line <- m_RNA_line %>% 
  group_by(line) %>% 
  summarise_all(concat_unique)
m_RNA_line <- m_RNA_line[match(ID_order, m_RNA_line$line),]

#match protein and RNA expression
prot <- d.log2[,match(m_prot$ID ,colnames(d.log2))]
prot <- prot %>% 
  filter(!rownames(prot) %in% c("FUK","MLK4","ZAK"))

RNA <- RNA_ex[,match(RNA_meta_MIB2$ID ,colnames(RNA_ex))]
colnames(RNA) <- RNA_meta_MIB2$Experiment
RNA <- RNA[match(rownames(prot) ,rownames(RNA)),]

#collapse by line
#RNA
c_RNA <- data.frame(t(RNA))
c_RNA$line <- RNA_meta_MIB2$line
c_RNA <- c_RNA %>% relocate(line, .before = PDPK1)

c_RNA <- c_RNA %>% 
  group_by(line) %>% 
  summarise(across(where(is.numeric), mean))

c_RNA <- data.frame((c_RNA))
c_RNA <- c_RNA[match(m_RNA_line$line, c_RNA$line),]
rownames(c_RNA) <- c_RNA$line
c_RNA$line <- NULL
c_RNA <- data.frame(t(c_RNA))
c_RNA[c_RNA == 0] <- NA
c_RNA$genes <- rownames(c_RNA)

#protein
c_prot <- data.frame(t(prot))
c_prot$line <- m_prot$line
c_prot <- c_prot %>% relocate(line, .before = PDPK1)

c_prot <- c_prot %>% 
  group_by(line) %>% 
  summarise(across(where(is.numeric), mean))

c_prot <- data.frame((c_prot))
c_prot <- c_prot[match(m_prot_line$line, c_prot$line),]
rownames(c_prot) <- c_prot$line
c_prot$line <- NULL
c_prot <- data.frame(t(c_prot))
c_prot[c_prot == 0] <- NA
c_prot$genes <- rownames(c_prot)

#convert to long form for all
c_RNA <- gather(c_RNA, line, RNA, P319T1:P806T1, factor_key=TRUE)
c_prot <- gather(c_prot, line, MIB, P319T1:P806T1, factor_key=TRUE)

#combined
c_RNA$MIB <- c_prot$MIB
c_RNA_prot <- c_RNA

c_classical8 <- c_RNA_prot[grepl("TNIK|ERN2|ROCK2|TAOK3|RIPK3|MAPK13|CHKA|FRK", c_RNA_prot$genes),]

c_basal8 <- c_RNA_prot[grepl("MARK1|PRKD1|EGFR|NUAK1|MAP3K4|AAK1|IKBKE|GSK3B", c_RNA_prot$genes),]

#RTKs
c_RTKs <- c_RNA_prot[grepl("CSF1R|EPHA2|EPHB2|EPHB6|EGFR|IGF1R|INSR|MET", c_RNA_prot$genes),]

```

```{r RNA-prot correlation plots, fig.height=8, fig.width=8, echo = FALSE, message = FALSE, error=TRUE}
#top 8 basal and classical
classical8 <- ggscatter(c_classical8, x = "RNA", y = "MIB",
          title = "Classical kinases",
          add = "reg.line",
          size = 2,
          color = "genes", palette = "jco",
          fullrange = TRUE
          
)+ stat_cor(aes(color = genes), label.x = -5, size = 7)+
  theme(legend.position = "right", legend.title = element_text(size=18),legend.text = element_text(size=18)) + font("xy.text", size = 18) + font("xy", size =18) + font("title", size = 20)


basal8 <- ggscatter(c_basal8, x = "RNA", y = "MIB",
          title = "Basal-like kinases",
          add = "reg.line",
          size = 2,
          color = "genes", palette = "jco",
          fullrange = TRUE
          
)+stat_cor(aes(color = genes), label.x = -5, size = 7)+
  theme(legend.position = "right", legend.title = element_text(size=18),legend.text = element_text(size=18)) + font("xy.text", size = 18) + font("xy", size =18) + font("title", size = 20)


png("plots/supp4d.png",width=9,height=6,units="in",res=500)
classical8
invisible(dev.off())

png("plots/supp4e.png",width=9,height=6,units="in",res=500)
basal8
invisible(dev.off())

classical8
basal8

#RTKs
RTK_scatter <- ggscatter(c_RTKs, x = "RNA", y = "MIB",
          add = "reg.line",
          size = 1,
          color = "genes", palette = c("CSF1R" = "#1F77B4",
                                      "EPHA2" = "#2CA02C",
                                      "EPHB2" = "#D62728",
                                      "EPHB6" = "#9467BD",
                                      "EGFR" = "#8C564B",
                                      "IGF1R" = "#AEC7E8",
                                      "INSR" = "#FFBB78",
                                      "MET" = "#98DF8A"),
          fullrange = TRUE
          
)+stat_cor(aes(color = genes), label.x = -5, size = 7)+
  theme(legend.position = "right", legend.title = element_text(size=18),legend.text = element_text(size=18)) + font("xy.text", size = 18) + font("xy", size =18)

png("plots/supp4c.png",width=8,height=7,units="in",res=500)
RTK_scatter
invisible(dev.off())

RTK_scatter

#overall corr
allkinase_scatter<- ggscatter(c_RNA_prot, x = "RNA", y = "MIB",
          add = "reg.line",
          size = 1,
          color = "line", palette = col_line,
          fullrange = TRUE,
          facet.by = "line",
          xlab = "log2(TPM)",
          ylab = "log2(LFQ)"
)+
  stat_cor(aes(color = line), label.x = -4, label.y = 32, size = 6, p.accuracy = 0.001, r.accuracy = 0.01)+
  theme(legend.position = "none")+
  theme(strip.text = element_text(
    size = 15)) + font("xy.text", size = 16) + font("xy", size =20) + facet_wrap(~ line, ncol = 5, nrow = 3)

png("plots/supp4b.png",width=13.5,height=5.5,units="in",res=500)
allkinase_scatter
invisible(dev.off())

allkinase_scatter
```

```{r primary proteomics data, echo = FALSE, include=FALSE}
##define all dfs
blca_tot=YehKinome$blca_tot
blca_phos=YehKinome$blca_phos
blca_m=YehKinome$blca_m
blca_key=YehKinome$blca_key
brca_tot=YehKinome$brca_tot
brca_phos=YehKinome$brca_phos
brca_m=YehKinome$brca_m
brca_key=YehKinome$brca_key
pdac_tot=YehKinome$pdac_tot
pdac_phos=YehKinome$pdac_phos
pdac_m=YehKinome$pdac_m
kinasekey=YehKinome$kinasekey

##CPTAC PDAC
#create metadata for PurIST subtypes
pdac_m_subtype <- pdac_m$subtype$Subtype %>% 
  mutate(basal = as.numeric(if_else(PurIST == "Basal-like", "1", "0"))) %>% 
  mutate(classical = as.numeric(if_else(PurIST == "Classical", "1", "0")))

#change . to - in names
colnames(pdac_phos) <- gsub("\\.","-", colnames(pdac_phos))

#pull out only kinases
pdac_tot2 <- pdac_tot[rownames(pdac_tot) %in% kinasekey$Gene,]
pdac_phos2 <- pdac_phos[rownames(pdac_phos) %in% kinasekey$Gene,]

#create df for boxplots
pdac_tot_all <- data.frame(t(pdac_tot2))
pdac_tot_all$subtype <- pdac_m_subtype$PurIST

pdac_phos_all <- data.frame(t(pdac_phos2))
pdac_phos_all$subtype <- pdac_m_subtype$PurIST

#define kinases for boxplots
pdac_RTK <- c("EGFR", "EPHA2", "IGF1R", "AXL", "MET")
pdac_RTK_phos <- c("EGFR", "EPHA2", "IGF1R", "AXL")
pdac_classical3 <- c("TNIK", "ERN2", "ROCK2")


##CPTAC BRCA
#add for limma metadata
brca_m2 <- brca_m %>% 
  mutate(Basal = as.numeric(if_else(PAM50 == "Basal", "1", "0")),
         Her2 = as.numeric(if_else(PAM50 == "Her2", "1", "0")),
         LumA = as.numeric(if_else(PAM50 == "LumA", "1", "0")),
         LumB = as.numeric(if_else(PAM50 == "LumB", "1", "0")),
         Normallike = as.numeric(if_else(PAM50 == "Normal-like", "1", "0")),
         .after = "PAM50") %>% 
  mutate(other = as.numeric(if_else(Basal == "1", "0", "1")),
         .after = "Normallike") %>% 
  mutate(basalother = if_else(PAM50 == "Basal", "Basal", "Other"),
         .after = "PAM50")
rownames(brca_m2) <- brca_m2$Sample.ID

brca_m3 <- brca_m2 %>% 
  filter(grepl("Basal|LumA|LumB", brca_m2$PAM50)) %>% 
  mutate(BasLum = ifelse(PAM50 == "Basal", "Basal", "Luminal"))

#pull out kinases
brca_kin <- brca_tot[brca_tot$geneSymbol %in% kinasekey$Gene,] 

brca_kin$geneSymbol <- ifelse(duplicated(brca_kin[,c("geneSymbol")])|duplicated(brca_kin[,c("geneSymbol")],fromLast = TRUE),
                      make.unique(brca_kin$geneSymbol),brca_kin$geneSymbol)

brca_kin <- brca_kin %>% 
  rownames_to_column(var = "row_names") %>% 
  select(-row_names) %>% 
  column_to_rownames(var = "geneSymbol") %>% 
  select(-(1:13))


brca_kin_phos <- brca_phos[brca_phos$geneSymbol %in% kinasekey$Gene,] 

brca_kin_phos$geneSymbol <- ifelse(duplicated(brca_kin_phos[,c("geneSymbol")])|duplicated(brca_kin_phos[,c("geneSymbol")],fromLast = TRUE),
                      make.unique(brca_kin_phos$geneSymbol),brca_kin_phos$geneSymbol)

brca_kin_phos <- brca_kin_phos %>% 
  rownames_to_column(var = "row_names") %>% 
  select(-row_names) %>% 
  column_to_rownames(var = "geneSymbol") %>% 
  select(-(1:18))


brca_kin3 <- brca_kin %>% 
  select(matches(rownames(brca_m3)))

brca_kin_phos3 <- brca_kin_phos %>% 
  select(matches(rownames(brca_m3)))

#PCA data
brca_pca_tot <- pca(na.omit(brca_kin3), metadata = brca_m3, removeVar = 0.1)
brca_pca_phos <- pca(na.omit(brca_kin_phos3), metadata = brca_m3, removeVar = 0.1)

#consensus heatmap
brca_d_heatmap <- na.omit(brca_kin3)
brca_d_heatmap_scaled <- data.frame(t(scale(t(brca_d_heatmap))))

col_ha_brca = columnAnnotation(
  PAM50 = as.matrix(brca_m3$PAM50),
  col = list(PAM50 = col_brca),
  simple_anno_size = unit(0.8, "cm"), height = unit(2, "cm"),
  annotation_legend_param = list(PAM50 = list(title = "PAM50", title_gp = gpar(fontsize = 16), 
                                                labels_gp = gpar(fontsize = 16))),
  annotation_name_gp= gpar(fontsize = 16))

datCol_brca <- as.dist(1-cor(as.matrix(brca_d_heatmap),method="pearson"))
datRow_brca <- as.dist(1-cor(t(as.matrix(brca_d_heatmap)),method="pearson"))

Colv_brca <- as.dendrogram(
  ConsensusClusterPlus::ConsensusClusterPlus(
    d = datCol_brca,
    seed = 1, pFeature = 1, pItem = 0.8,
    maxK = 4, reps=500, distance="euclidean",
    clusterAlg="km")[[3]]$consensusTree)

Rowv_brca <- as.dendrogram(
  ConsensusClusterPlus::ConsensusClusterPlus(
    d = datRow_brca,
    seed = 1, pFeature = 1, pItem = 0.8,
    maxK = 4, reps=500, distance="euclidean",
    clusterAlg="km")[[3]]$consensusTree)

#df for boxplots
brca_tot_all <- data.frame(t(brca_kin))
brca_tot_all$PAM50 <- brca_m2$PAM50
brca_tot_all$subtype <- brca_m2$basalother
brca_tot_all <- brca_tot_all %>% 
  relocate(PAM50, .before = TTN) %>% 
  relocate(subtype, .before = PAM50)

brca_phos_all <- data.frame(t(brca_kin_phos))
brca_phos_all$PAM50 <- brca_m2$PAM50
brca_phos_all$subtype <- brca_m2$basalother
brca_phos_all <- brca_phos_all %>% 
  relocate(PAM50, .before = RPS6KA1) %>% 
  relocate(subtype, .before = PAM50)

brca_tot_all2 <- brca_tot_all %>% 
  filter(!grepl("Her2|Normal", brca_tot_all$PAM50))

brca_phos_all2 <- brca_phos_all %>% 
  filter(!grepl("Her2|Normal", brca_phos_all$PAM50))


##BLCA
#parse data
blca_m2 <- blca_m %>% 
  left_join(blca_key, by = "Case.ID") %>% 
  relocate(UNC, .after = RNA.seq.ID) %>% 
  relocate(Subtype.Consnsus, .after = RNA.seq.ID) %>% 
  mutate(basal = as.numeric(ifelse(UNC == "Basal", "1", "0")),
         luminal = as.numeric(ifelse(UNC == "Luminal", "1", "0")),
         ba_sq = as.numeric(ifelse(Subtype.Consnsus == "Ba/Sq", "1", "0")),
         lumP = as.numeric(ifelse(Subtype.Consnsus == "LumP", "1", "0")),
         .after = Pathology.reviewed) %>% 
  filter(!is.na(UNC)) %>% 
  mutate_all(~ ifelse(. == "", NA_character_, .))

blca_m_tot <- blca_m2 %>% 
  filter(!is.na(Proteome.ID)) %>%
  column_to_rownames(var = "Proteome.ID") 

blca_m_phos <- blca_m2 %>% 
  filter(!is.na(Phosphoproteome.ID)) %>%
  column_to_rownames(var = "Phosphoproteome.ID")

blca_m_tot2 <- blca_m_tot %>% 
  filter(grepl("Ba/Sq|LumP", blca_m_tot$Subtype.Consnsus))

blca_m_phos2 <- blca_m_phos %>% 
  filter(grepl("Ba/Sq|LumP", blca_m_phos$Subtype.Consnsus))

#pull kinases
blca_tot_kin <- blca_tot[row.names(blca_tot) %in% kinasekey$Gene,] %>% 
  select(matches(rownames(blca_m_tot))) %>% 
  rename_with(~ str_remove(., "_.*"), everything()) %>%
  mutate(across(everything(), ~ log2(.)))

blca_phos_kin <- blca_phos[row.names(blca_phos) %in% kinasekey$Gene,] %>% 
  select(matches(rownames(blca_m_phos))) %>% 
  rename_with(~ str_remove(., "_.*"), everything()) %>%
  mutate(across(everything(), ~ log2(.)))


blca_tot_kin2 <- blca_tot_kin %>% 
    select(matches(rownames(blca_m_tot2)))

blca_phos_kin2 <- blca_phos_kin %>% 
    select(matches(rownames(blca_m_phos2)))

#PCA
blca_pca2 <- pca(na.omit(blca_tot_kin2), metadata = blca_m_tot2, removeVar = 0.1)

#heatmap
blca_d_heatmap2 <- na.omit(blca_tot_kin2)
blca_d_heatmap_scaled2 <- data.frame(t(scale(t(blca_d_heatmap2))))

col_ha_blca = columnAnnotation(
  UNC = as.matrix(blca_m_tot2$UNC),
  consensus = as.matrix(blca_m_tot2$Subtype.Consnsus),
  col = list(UNC = c("LumA" = "lightblue", 
                  "Luminal" = "blue", 
                  "Her2" = "purple", 
                  "Basal" = "#fc8803", 
                  "Normal-like" = "grey"),
             consensus = c("LumU" = "lightblue", 
                  "LumP" = "blue", 
                  "LumNS" = "purple", 
                  "Ba/Sq" = "#fc8803", 
                  "Stroma-rich" = "grey",
                  "NE-like" = "black")),
  simple_anno_size = unit(0.8, "cm"), height = unit(2, "cm"),
  annotation_name_gp= gpar(fontsize = 16))

datCol_blca2 <- as.dist(1-cor(as.matrix(blca_d_heatmap2),method="pearson"))
datRow_blca2 <- as.dist(1-cor(t(as.matrix(blca_d_heatmap2)),method="pearson"))

Colv_blca2 <- as.dendrogram(
  ConsensusClusterPlus::ConsensusClusterPlus(
    d = datCol_blca2,
    seed = 1, pFeature = 1, pItem = 0.8,
    maxK = 3, reps=500, distance="euclidean",
    clusterAlg="km")[[2]]$consensusTree)

Rowv_blca2 <- as.dendrogram(
  ConsensusClusterPlus::ConsensusClusterPlus(
    d = datRow_blca2,
    seed = 1, pFeature = 1, pItem = 0.8,
    maxK = 3, reps=500, distance="euclidean",
    clusterAlg="km")[[2]]$consensusTree)

#boxplots
all_blca2 <- data.frame(t(blca_tot_kin2))
all_blca2$UNC <- blca_m_tot2$UNC
all_blca2$consensus <- blca_m_tot2$Subtype.Consnsus
all_blca2 <- all_blca2 %>% 
  relocate(UNC, .before = NME2) %>% 
  relocate(consensus, .before = UNC)

all_phos_blca2 <- data.frame(t(blca_phos_kin2))
all_phos_blca2$UNC <- blca_m_phos2$UNC
all_phos_blca2$consensus <- blca_m_phos2$Subtype.Consnsus
all_phos_blca2 <- all_phos_blca2 %>% 
  relocate(UNC, .before = AAK1) %>% 
  relocate(consensus, .before = UNC)

```


```{r primary proteomics boxplots PDAC, fig.height=3, fig.width=11, echo = FALSE, message = FALSE, error=TRUE}
#CPTAC PDAC
my_plot_list = list()
for(i in 1:5){
  p <- ggboxplot(pdac_tot_all, x = "subtype", y = pdac_RTK[i], 
                 color = "subtype", palette = c("#0331fc","#fc8803"), 
                 legend = "none", 
                 title = paste(pdac_RTK[i],"total", sep = "_"),
                 font.main = c(12, "bold", "black"),
                 font.x = 12,
                 font.xtickslab = 12,
                 ylab = FALSE,
                 add.params = list(size = 0.5),
                 add = "jitter") + stat_compare_means(size = 4, aes(label = paste0("p = ", ..p.format..)),
                                                      method = "t.test",
                 ) +  scale_y_continuous(expand = expansion(mult = c(0, 0.1)))
  my_plot_list[[i]] <- p
}

pdac_RTK_tot_boxplots <- ggarrange(plotlist = my_plot_list,  nrow = 1, ncol = 5)
annotate_figure(pdac_RTK_tot_boxplots, left = text_grob("Log2(LFQ)", color = "black",face = "bold", rot = 90))


my_plot_list = list()
for(i in 1:4){
  p <- ggboxplot(pdac_phos_all, x = "subtype", y = pdac_RTK_phos[i], 
                 color = "subtype", palette = c("#0331fc","#fc8803"), 
                 legend = "none", 
                 title = paste(pdac_RTK_phos[i],"phospho", sep = "_"),
                 font.main = c(12, "bold", "black"),
                 font.x = 12,
                 font.xtickslab = 12,
                 ylab = FALSE,
                 add.params = list(size = 0.5),
                 add = "jitter") + stat_compare_means(size = 4, aes(label = paste0("p = ", ..p.format..)),
                                                      method = "t.test",
                 ) +  scale_y_continuous(expand = expansion(mult = c(0, 0.1)))
  my_plot_list[[i]] <- p
}

pdac_RTK_phos_boxplots <- ggarrange(plotlist = my_plot_list,  nrow = 1, ncol = 5)
annotate_figure(pdac_RTK_phos_boxplots, left = text_grob("Log2(LFQ)", color = "black",face = "bold", rot = 90))


my_plot_list = list()
for(i in 1:3){
  p <- ggboxplot(pdac_tot_all, x = "subtype", y = pdac_classical3[i], 
                 color = "subtype", palette = c("#0331fc","#fc8803"), 
                 legend = "none", 
                 title = paste(pdac_classical3[i],"total", sep = "_"),
                 font.main = c(12, "bold", "black"),
                 font.x = 12,
                 font.xtickslab = 12,
                 ylab = FALSE,
                 add.params = list(size = 0.5),
                 add = "jitter") + stat_compare_means(size = 4, aes(label = paste0("p = ", ..p.format..)),
                                                      method = "t.test",
                 ) +  scale_y_continuous(expand = expansion(mult = c(0, 0.1)))
  my_plot_list[[i]] <- p
}

pdac_classical_tot_boxplots <- ggarrange(plotlist = my_plot_list,  nrow = 1, ncol = 3)
annotate_figure(pdac_classical_tot_boxplots, left = text_grob("Log2(LFQ)", color = "black",face = "bold", rot = 90))

png("plots/supp4a_1.png",width=11,height=3,units="in",res=500)
pdac_RTK_tot_boxplots
invisible(dev.off())

png("plots/supp4a_2.png",width=11,height=3,units="in",res=500)
pdac_RTK_phos_boxplots
invisible(dev.off())

png("plots/supp4b.png",width=7,height=3,units="in",res=500)
pdac_classical_tot_boxplots
invisible(dev.off())
```

```{r primary proteomics boxplots BRCA and BLCA, fig.height=3, fig.width=4.5, echo = FALSE, message = FALSE, error=TRUE}
#brca
brca_EGFR_tot <- ggboxplot(brca_tot_all2, x = "subtype", y = "EGFR", 
          color = "subtype", palette = c("#fc8803", "#0331fc"), legend = "none", title = "EGFR total",
          add.params = list(size = 0.5),
          font.main = c(12, "bold", "black"),
          font.x = 12,
          font.xtickslab = 12,
          ylab = FALSE,
          add = "jitter") + stat_compare_means(size = 5, aes(label = paste0("p =", ..p.format..)),
                                               method = "t.test") +  scale_y_continuous(expand = expansion(mult = c(0, 0.1)))

brca_EGFR_phos <- ggboxplot(brca_phos_all2, x = "subtype", y = "EGFR", 
          color = "subtype", palette = c("#fc8803", "#0331fc"), legend = "none", title = "EGFR phospho",
          add.params = list(size = 0.5),
          font.main = c(12, "bold", "black"),
          font.x = 12,
          font.xtickslab = 12,
          ylab = FALSE,
          add = "jitter") + stat_compare_means(size = 5, aes(label = paste0("p =", ..p.format..)),
                                               method = "t.test") +  scale_y_continuous(expand = expansion(mult = c(0, 0.1)))

brca_boxplots <- ggarrange(brca_EGFR_tot, brca_EGFR_phos , nrow = 1, ncol = 2)
annotate_figure(brca_boxplots,
                left = text_grob("Normalized Intensity", color = "black",face = "bold", rot = 90))

png("plots/supp5i.png",width=4.5,height=3,units="in",res=500)
annotate_figure(brca_boxplots,
                left = text_grob("Normalized Intensity", color = "black",face = "bold", rot = 90))
invisible(dev.off())

#blca
EGFR_tot_blca <- ggboxplot(all_blca2, x = "consensus", y = "EGFR", 
          color = "consensus", palette = c("#fc8803", "#0331fc"), legend = "none", title = "EGFR total",
          add.params = list(size = 0.5),
          font.main = c(12, "bold", "black"),
          font.x = 12,
          font.xtickslab = 12,
          ylab = FALSE,
          add = "jitter") + stat_compare_means(size = 5, aes(label = paste0("p =", ..p.format..)),
                                               method = "t.test") +  scale_y_continuous(expand = expansion(mult = c(0, 0.1)))

EGFR_phos_blca <- ggboxplot(all_phos_blca2, x = "consensus", y = "EGFR", 
          color = "consensus", palette = c("#fc8803", "#0331fc"), legend = "none", title = "EGFR phospho",
          add.params = list(size = 0.5),
          font.main = c(12, "bold", "black"),
          font.x = 12,
          font.xtickslab = 12,
          ylab = FALSE,
          add = "jitter") + stat_compare_means(size = 5, aes(label = paste0("p =", ..p.format..)),
                                               method = "t.test") +  scale_y_continuous(expand = expansion(mult = c(0, 0.1)))

png("plots/supp5f.png",width=4.5,height=3,units="in",res=500)
blca_boxplots <- ggarrange(EGFR_tot_blca, EGFR_phos_blca , nrow = 1, ncol = 2)
annotate_figure(blca_boxplots,
                left = text_grob("Normalized Intensity", color = "black",face = "bold", rot = 90))
invisible(dev.off())

annotate_figure(blca_boxplots,
                left = text_grob("Normalized Intensity", color = "black",face = "bold", rot = 90))
```

```{r primary proteomics PCA plots, fig.height=5, fig.width=6, echo = FALSE, message = FALSE, error=TRUE}
brca_pca_plot <- biplot(brca_pca_tot,
       showLoadings = F,
       lengthLoadingsArrowsFactor = 1.5,
       x = 'PC1', y = 'PC2',
       sizeLoadingsNames = 5,
       lab = NULL,
       title = "CPTAC BRCA total proteome",
       colby = "PAM50", colLegendTitle = "PAM50",
       colkey = col_brca,
       pointSize = 1.5,
       legendPosition = 'top',
       legendLabSize = 16, legendIconSize = 3, legendTitleSize = 16,
       axisLabSize = 16)

brca_pca_phos_plot <-biplot(brca_pca_phos,
       showLoadings = F,
       lengthLoadingsArrowsFactor = 1.5,
       x = 'PC1', y = 'PC2',
       sizeLoadingsNames = 5,
       lab = NULL,
       title = "CPTAC BRCA phospho proteome",
       colby = "PAM50", colLegendTitle = "PAM50",
       colkey = col_brca,
       pointSize = 1.5,
       legendPosition = 'top',
       legendLabSize = 16, legendIconSize = 3, legendTitleSize = 16,
       axisLabSize = 16)

png("plots/supp5h_1.png",width=6,height=5,units="in",res=500)
brca_pca_plot
invisible(dev.off())

png("plots/supp5h_2.png",width=6,height=5,units="in",res=500)
brca_pca_phos_plot
invisible(dev.off())

brca_pca_plot
brca_pca_phos_plot

#blca
blca_pca_plot <- biplot(blca_pca2,
       showLoadings = F,
       lengthLoadingsArrowsFactor = 1.5,
       x = 'PC1', y = 'PC2',
       sizeLoadingsNames = 5,
       lab = NULL,
       title = "CPTAC BLCA total proteome",
       colby = "UNC", colLegendTitle = "UNC subtypes",
       colkey = c("Luminal" = "blue", 
                  "Basal" = "#fc8803"),
       pointSize = 1.5,
       legendPosition = 'right',
       legendLabSize = 16, legendIconSize = 3, legendTitleSize = 16,
       axisLabSize = 16)

png("plots/supp5e.png",width=6,height=5,units="in",res=500)
blca_pca_plot
invisible(dev.off())

blca_pca_plot

```

```{r primary proteomics heatmaps, fig.height=8, fig.width=9, echo = FALSE, message = FALSE, error=TRUE}
#blca
ht_opt(RESET = TRUE)
ht_opt(legend_title_gp = gpar(fontsize = 16), 
       legend_labels_gp = gpar(fontsize = 16))

htmap_blca <- Heatmap(as.matrix(blca_d_heatmap_scaled2),
                 name = "LFQ",
                 col = col_fun_htmp,
                 cluster_columns = Colv_blca2,
                 cluster_rows = Rowv_blca2,
                 show_row_names = F,
                 show_column_names = FALSE,
                 column_names_gp = gpar(fontsize = 6),
                 top_annotation = col_ha_blca,
                 heatmap_legend_param = list(
                   at = c(-4, 0, 4),
                   title = "intensity",
                   legend_height = unit(4, "cm"),
                   title_position = "topcenter"))


draw(htmap_blca, heatmap_legend_side="left", annotation_legend_side="left",
     legend_grouping = "original")

png("plots/supp5d.png",width=9,height=8,units="in",res=500)
draw(htmap_blca, heatmap_legend_side="left", annotation_legend_side="left",
     legend_grouping = "original")
invisible(dev.off())

#brca
htmp_brca <- Heatmap(as.matrix(brca_d_heatmap_scaled),
                 name = "LFQ",
                 col = col_fun_htmp,
                 cluster_columns = Colv_brca,
                 cluster_rows = Rowv_brca,
                 show_row_names = F,
                 show_column_names = FALSE,
                 column_title = "total protein k = 3",
                 column_names_gp = gpar(fontsize = 6),
                 top_annotation = col_ha_brca,
                 heatmap_legend_param = list(
                   at = c(-4, 0, 4),
                   title = "intensity",
                   legend_height = unit(4, "cm"),
                   title_position = "topcenter")
)

draw(htmp_brca, heatmap_legend_side="left", annotation_legend_side="left",
     legend_grouping = "original")

png("plots/supp5g.png",width=7,height=5,units="in",res=500)
draw(htmp_brca, heatmap_legend_side="left", annotation_legend_side="left",
     legend_grouping = "original")
invisible(dev.off())
```

